{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "from copy import copy\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get HGNC Mapper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(21435, 5)\n",
      "Index(['HGNC ID', 'Approved symbol', 'Approved name', 'Previous symbols',\n",
      "       'Synonyms'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC ID</th>\n",
       "      <th>Approved symbol</th>\n",
       "      <th>Approved name</th>\n",
       "      <th>Previous symbols</th>\n",
       "      <th>Synonyms</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HGNC:37133</td>\n",
       "      <td>A1BG-AS1</td>\n",
       "      <td>A1BG antisense RNA 1</td>\n",
       "      <td>NCRNA00181, A1BGAS, A1BG-AS</td>\n",
       "      <td>FLJ23569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HGNC:24086</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>APOBEC1 complementation factor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ACF, ASP, ACF64, ACF65, APOBEC1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HGNC:7</td>\n",
       "      <td>A2M</td>\n",
       "      <td>alpha-2-macroglobulin</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FWP007, S863-7, CPAMD5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HGNC:23336</td>\n",
       "      <td>A2ML1</td>\n",
       "      <td>alpha-2-macroglobulin like 1</td>\n",
       "      <td>CPAMD9</td>\n",
       "      <td>FLJ25179, p170</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HGNC:30005</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>alpha 1,3-galactosyltransferase 2</td>\n",
       "      <td>A3GALT2P</td>\n",
       "      <td>IGBS3S, IGB3S</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       HGNC ID Approved symbol                      Approved name  \\\n",
       "1   HGNC:37133        A1BG-AS1               A1BG antisense RNA 1   \n",
       "2   HGNC:24086            A1CF     APOBEC1 complementation factor   \n",
       "4       HGNC:7             A2M              alpha-2-macroglobulin   \n",
       "6   HGNC:23336           A2ML1       alpha-2-macroglobulin like 1   \n",
       "12  HGNC:30005         A3GALT2  alpha 1,3-galactosyltransferase 2   \n",
       "\n",
       "               Previous symbols                           Synonyms  \n",
       "1   NCRNA00181, A1BGAS, A1BG-AS                           FLJ23569  \n",
       "2                           NaN  ACF, ASP, ACF64, ACF65, APOBEC1CF  \n",
       "4                           NaN             FWP007, S863-7, CPAMD5  \n",
       "6                        CPAMD9                     FLJ25179, p170  \n",
       "12                     A3GALT2P                      IGBS3S, IGB3S  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(11702, 5)\n",
      "Index(['HGNC ID', 'Approved symbol', 'Approved name', 'Previous symbols',\n",
      "       'Synonyms'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC ID</th>\n",
       "      <th>Approved symbol</th>\n",
       "      <th>Approved name</th>\n",
       "      <th>Previous symbols</th>\n",
       "      <th>Synonyms</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HGNC:37133</td>\n",
       "      <td>A1BG-AS1</td>\n",
       "      <td>A1BG antisense RNA 1</td>\n",
       "      <td>NCRNA00181, A1BGAS, A1BG-AS</td>\n",
       "      <td>FLJ23569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HGNC:23336</td>\n",
       "      <td>A2ML1</td>\n",
       "      <td>alpha-2-macroglobulin like 1</td>\n",
       "      <td>CPAMD9</td>\n",
       "      <td>FLJ25179, p170</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HGNC:8</td>\n",
       "      <td>A2MP1</td>\n",
       "      <td>alpha-2-macroglobulin pseudogene 1</td>\n",
       "      <td>A2MP</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HGNC:30005</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>alpha 1,3-galactosyltransferase 2</td>\n",
       "      <td>A3GALT2P</td>\n",
       "      <td>IGBS3S, IGB3S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>HGNC:18149</td>\n",
       "      <td>A4GALT</td>\n",
       "      <td>alpha 1,4-galactosyltransferase (P blood group)</td>\n",
       "      <td>P1</td>\n",
       "      <td>A14GALT, Gb3S, P(k)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       HGNC ID Approved symbol  \\\n",
       "1   HGNC:37133        A1BG-AS1   \n",
       "6   HGNC:23336           A2ML1   \n",
       "9       HGNC:8           A2MP1   \n",
       "12  HGNC:30005         A3GALT2   \n",
       "13  HGNC:18149          A4GALT   \n",
       "\n",
       "                                      Approved name  \\\n",
       "1                              A1BG antisense RNA 1   \n",
       "6                      alpha-2-macroglobulin like 1   \n",
       "9                alpha-2-macroglobulin pseudogene 1   \n",
       "12                alpha 1,3-galactosyltransferase 2   \n",
       "13  alpha 1,4-galactosyltransferase (P blood group)   \n",
       "\n",
       "               Previous symbols             Synonyms  \n",
       "1   NCRNA00181, A1BGAS, A1BG-AS             FLJ23569  \n",
       "6                        CPAMD9       FLJ25179, p170  \n",
       "9                          A2MP                  NaN  \n",
       "12                     A3GALT2P        IGBS3S, IGB3S  \n",
       "13                           P1  A14GALT, Gb3S, P(k)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "hgnc = pd.read_csv('../data/hgnc_alias_list.txt', sep='\\t')\n",
    "#hgnc.head()\n",
    "\n",
    "hgnc = hgnc[hgnc['Approved symbol'].apply(lambda x: 'withdrawn' not in x)]\n",
    "\n",
    "# get the original keys\n",
    "hgnc_original_keys = hgnc['Approved symbol'].unique()\n",
    "\n",
    "# drop a lot of columns for efficiency\n",
    "hgnc = hgnc[list(hgnc.columns)[0:6]].drop('Status', axis=1)\n",
    "\n",
    "# filter out Nan synonyms (not helpful)\n",
    "hgnc_syn_list = hgnc[~ hgnc.Synonyms.isna()]\n",
    "hgnc_prev_symb_list = hgnc[~ hgnc['Previous symbols'].isna()]\n",
    "\n",
    "print(hgnc_syn_list.shape)\n",
    "print(hgnc_syn_list.columns)\n",
    "display(hgnc_syn_list.head())\n",
    "\n",
    "print(hgnc_prev_symb_list.shape)\n",
    "print(hgnc_prev_symb_list.columns)\n",
    "display(hgnc_prev_symb_list.head())\n",
    "\n",
    "#convert the synonyms column to a list\n",
    "# convert these lists to pd.Series\n",
    "# merge with original dataframe\n",
    "#drop old synonyms column \n",
    "# melt the new columns into rows\n",
    "\n",
    "\n",
    "current_syn_list = hgnc_syn_list.Synonyms.apply(lambda x: x.split(',')) \\\n",
    "    .apply(pd.Series) \\\n",
    "    .merge(hgnc, left_index = True, right_index = True) \\\n",
    "    .drop([\"Synonyms\"], axis = 1) \\\n",
    "    .melt(id_vars = ['HGNC ID', 'Approved symbol', 'Approved name', 'Previous symbols'], value_name = \"synonym\") \n",
    "\n",
    "current_syn_list = current_syn_list[~ current_syn_list.synonym.isna()]\n",
    "current_syn_list.synonym = current_syn_list.synonym.apply(lambda x: x.replace(' ',''))\n",
    "\n",
    "\n",
    "current_syn_list.head()\n",
    "\n",
    "prev_symb_list = hgnc_prev_symb_list['Previous symbols'].apply(lambda x: x.split(',')) \\\n",
    "    .apply(pd.Series) \\\n",
    "    .merge(hgnc, left_index = True, right_index = True) \\\n",
    "    .drop(['Previous symbols'], axis = 1) \\\n",
    "    .melt(id_vars = ['HGNC ID', 'Approved symbol', 'Approved name', 'Synonyms'], value_name = \"synonym\") \n",
    "\n",
    "prev_symb_list = prev_symb_list[~ prev_symb_list.synonym.isna()]\n",
    "prev_symb_list.synonym = prev_symb_list.synonym.apply(lambda x: x.replace(' ',''))\n",
    "\n",
    "\n",
    "prev_symb_list.head()\n",
    "\n",
    "hgnc_mapper = dict(zip(current_syn_list['synonym'], current_syn_list['Approved symbol']))\n",
    "# add in HGNC ID mapper\n",
    "hgnc_mapper.update(dict(zip(current_syn_list['HGNC ID'], current_syn_list['Approved symbol'])))\n",
    "hgnc_mapper_previous = dict(zip(prev_symb_list['synonym'], prev_symb_list['Approved symbol']))\n",
    "\n",
    "trouble_list = list(filter(lambda x: hgnc_mapper[x] != hgnc_mapper_previous[x], set(hgnc_mapper.keys())&set(hgnc_mapper_previous.keys())))\n",
    "\n",
    "hand_coded = {'RAGE':'MOK', 'SGK2':'SGK2', 'SGK196':'SGK196', 'MAPK3':'MAPK3'}\n",
    "\n",
    "hgnc_mapper_previous.update(hgnc_mapper) #overwrite the previous symbol conflicts\n",
    "\n",
    "hgnc_mapper = hgnc_mapper_previous\n",
    "hgnc_mapper.update({x:x for x in hgnc_original_keys}) #keep the identify maps\n",
    "hgnc_mapper.update(hand_coded) # overwrite the trouble list"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get Uniprot mapper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('P31946', 'YWHAB')"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uni_mapper = pd.read_csv('../data/network/HUMAN_9606_idmapping.dat', sep='\\t', header=None)\n",
    "uni_mapper = uni_mapper[uni_mapper[1] == 'Gene_Name'].set_index(0)[2].to_dict()\n",
    "\n",
    "next(iter(uni_mapper.items()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "uni_rev = {y:x for x,y in uni_mapper.items()}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "146363\n",
      "26468\n"
     ]
    }
   ],
   "source": [
    "print(len(uni_mapper))\n",
    "print(len(uni_rev))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Extra mappers from UniProt, e.g. ENSG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(169389, 22)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>12</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "      <th>15</th>\n",
       "      <th>16</th>\n",
       "      <th>17</th>\n",
       "      <th>18</th>\n",
       "      <th>19</th>\n",
       "      <th>20</th>\n",
       "      <th>21</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P31946</td>\n",
       "      <td>1433B_HUMAN</td>\n",
       "      <td>7529</td>\n",
       "      <td>NP_003395.1; NP_647539.1; XP_016883528.1</td>\n",
       "      <td>4507949; 377656702; 67464628; 1345590; 1034625...</td>\n",
       "      <td>2BQ0:A; 2BQ0:B; 2C23:A; 4DNK:A; 4DNK:B; 5N10:A...</td>\n",
       "      <td>GO:0005737; GO:0005829; GO:0070062; GO:0005925...</td>\n",
       "      <td>UniRef100_P31946</td>\n",
       "      <td>UniRef90_P31946</td>\n",
       "      <td>UniRef50_P31946</td>\n",
       "      <td>...</td>\n",
       "      <td>9606</td>\n",
       "      <td>601289</td>\n",
       "      <td>Hs.643544</td>\n",
       "      <td>8515476; 14702039; 11780052; 15489334; 2357255...</td>\n",
       "      <td>X57346; AK292717; AL008725; CH471077; CH471077...</td>\n",
       "      <td>CAA40621.1; BAF85406.1; -; EAW75893.1; EAW7589...</td>\n",
       "      <td>ENSG00000166913</td>\n",
       "      <td>ENST00000353703; ENST00000372839</td>\n",
       "      <td>ENSP00000300161; ENSP00000361930</td>\n",
       "      <td>11996670; 12364343; 12437930; 12468542; 124825...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P62258</td>\n",
       "      <td>1433E_HUMAN</td>\n",
       "      <td>7531</td>\n",
       "      <td>NP_006752.1</td>\n",
       "      <td>374074368; 62131678; 67464424; 194379794; 2210...</td>\n",
       "      <td>2BR9:A; 3UAL:A; 3UBW:A; 6EIH:A</td>\n",
       "      <td>GO:0090724; GO:0005737; GO:0005829; GO:0070062...</td>\n",
       "      <td>UniRef100_P62258</td>\n",
       "      <td>UniRef90_P62258</td>\n",
       "      <td>UniRef50_P62258</td>\n",
       "      <td>...</td>\n",
       "      <td>9606</td>\n",
       "      <td>605066</td>\n",
       "      <td>Hs.513851</td>\n",
       "      <td>7644510; 8858348; 8684458; 20417184; 14702039;...</td>\n",
       "      <td>U20972; U54778; U43399; U43430; U28936; AB0171...</td>\n",
       "      <td>AAC50175.1; AAC50710.1; AAC50625.1; AAD00026.1...</td>\n",
       "      <td>ENSG00000108953; ENSG00000274474</td>\n",
       "      <td>ENST00000264335; ENST00000571732; ENST00000616...</td>\n",
       "      <td>ENSP00000264335; ENSP00000461762; ENSP00000481...</td>\n",
       "      <td>15838597; 11782387; 12657644; 14966136; 153649...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Q04917</td>\n",
       "      <td>1433F_HUMAN</td>\n",
       "      <td>7533</td>\n",
       "      <td>NP_003396.1</td>\n",
       "      <td>4507951; 437363; 83754686; 83754699; 83754700;...</td>\n",
       "      <td>2C63:A; 2C63:B; 2C63:C; 2C63:D; 2C74:A; 2C74:B</td>\n",
       "      <td>GO:0005737; GO:0005829; GO:0070062; GO:0098978...</td>\n",
       "      <td>UniRef100_Q04917</td>\n",
       "      <td>UniRef90_Q04917</td>\n",
       "      <td>UniRef50_P62258</td>\n",
       "      <td>...</td>\n",
       "      <td>9606</td>\n",
       "      <td>113508</td>\n",
       "      <td>Hs.226755</td>\n",
       "      <td>8218406; 1578511; 8561965; 8812417; 15461802; ...</td>\n",
       "      <td>L20422; X80536; X78138; X57345; D78577; S80794...</td>\n",
       "      <td>AAA35483.1; CAA56676.1; CAA55017.1; CAA40620.1...</td>\n",
       "      <td>ENSG00000128245</td>\n",
       "      <td>ENST00000248975</td>\n",
       "      <td>ENSP00000248975</td>\n",
       "      <td>10206237; 11121172; 11996670; 12176995; 124801...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P61981</td>\n",
       "      <td>1433G_HUMAN</td>\n",
       "      <td>7532</td>\n",
       "      <td>NP_036611.2</td>\n",
       "      <td>6016838; 380764684; 635576381; 82407956; 82407...</td>\n",
       "      <td>2B05:A; 2B05:B; 2B05:C; 2B05:D; 2B05:E; 2B05:F...</td>\n",
       "      <td>GO:0005829; GO:0070062; GO:0005925; GO:0016020...</td>\n",
       "      <td>UniRef100_P61981</td>\n",
       "      <td>UniRef90_P61981</td>\n",
       "      <td>UniRef50_P61981</td>\n",
       "      <td>...</td>\n",
       "      <td>9606</td>\n",
       "      <td>605356; 617665</td>\n",
       "      <td>Hs.744840</td>\n",
       "      <td>10433554; 10486217; 12853948; 15489334; 235725...</td>\n",
       "      <td>AF142498; AB024334; CR541904; CR541925; AC0063...</td>\n",
       "      <td>AAD48408.1; BAA85184.1; CAG46702.1; CAG46723.1...</td>\n",
       "      <td>ENSG00000170027</td>\n",
       "      <td>ENST00000307630</td>\n",
       "      <td>ENSP00000306330</td>\n",
       "      <td>11824616; 11996670; 12364343; 12482592; 150572...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P31947</td>\n",
       "      <td>1433S_HUMAN</td>\n",
       "      <td>2810</td>\n",
       "      <td>NP_006133.1</td>\n",
       "      <td>436408756; 306991738; 969812714; 350610438; 35...</td>\n",
       "      <td>1YWT:A; 1YWT:B; 1YZ5:A; 1YZ5:B; 3IQJ:A; 3IQU:A...</td>\n",
       "      <td>GO:0005829; GO:0070062; GO:0005615; GO:0005739...</td>\n",
       "      <td>UniRef100_P31947</td>\n",
       "      <td>UniRef90_P31947</td>\n",
       "      <td>UniRef50_P61981</td>\n",
       "      <td>...</td>\n",
       "      <td>9606</td>\n",
       "      <td>601290</td>\n",
       "      <td>Hs.523718</td>\n",
       "      <td>1390337; 8515476; 9659898; 16710414; 15489334;...</td>\n",
       "      <td>M93010; X57348; AF029081; AF029082; CR541905; ...</td>\n",
       "      <td>AAA59546.1; CAA40623.1; AAC52029.1; AAC52030.1...</td>\n",
       "      <td>ENSG00000175793</td>\n",
       "      <td>ENST00000339276</td>\n",
       "      <td>ENSP00000340989</td>\n",
       "      <td>10969776; 12582028; 12730237; 12787309; 145172...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       0            1     2                                         3   \\\n",
       "0  P31946  1433B_HUMAN  7529  NP_003395.1; NP_647539.1; XP_016883528.1   \n",
       "1  P62258  1433E_HUMAN  7531                               NP_006752.1   \n",
       "2  Q04917  1433F_HUMAN  7533                               NP_003396.1   \n",
       "3  P61981  1433G_HUMAN  7532                               NP_036611.2   \n",
       "4  P31947  1433S_HUMAN  2810                               NP_006133.1   \n",
       "\n",
       "                                                  4   \\\n",
       "0  4507949; 377656702; 67464628; 1345590; 1034625...   \n",
       "1  374074368; 62131678; 67464424; 194379794; 2210...   \n",
       "2  4507951; 437363; 83754686; 83754699; 83754700;...   \n",
       "3  6016838; 380764684; 635576381; 82407956; 82407...   \n",
       "4  436408756; 306991738; 969812714; 350610438; 35...   \n",
       "\n",
       "                                                  5   \\\n",
       "0  2BQ0:A; 2BQ0:B; 2C23:A; 4DNK:A; 4DNK:B; 5N10:A...   \n",
       "1                     2BR9:A; 3UAL:A; 3UBW:A; 6EIH:A   \n",
       "2     2C63:A; 2C63:B; 2C63:C; 2C63:D; 2C74:A; 2C74:B   \n",
       "3  2B05:A; 2B05:B; 2B05:C; 2B05:D; 2B05:E; 2B05:F...   \n",
       "4  1YWT:A; 1YWT:B; 1YZ5:A; 1YZ5:B; 3IQJ:A; 3IQU:A...   \n",
       "\n",
       "                                                  6                 7   \\\n",
       "0  GO:0005737; GO:0005829; GO:0070062; GO:0005925...  UniRef100_P31946   \n",
       "1  GO:0090724; GO:0005737; GO:0005829; GO:0070062...  UniRef100_P62258   \n",
       "2  GO:0005737; GO:0005829; GO:0070062; GO:0098978...  UniRef100_Q04917   \n",
       "3  GO:0005829; GO:0070062; GO:0005925; GO:0016020...  UniRef100_P61981   \n",
       "4  GO:0005829; GO:0070062; GO:0005615; GO:0005739...  UniRef100_P31947   \n",
       "\n",
       "                8                9   ...    12              13         14  \\\n",
       "0  UniRef90_P31946  UniRef50_P31946  ...  9606          601289  Hs.643544   \n",
       "1  UniRef90_P62258  UniRef50_P62258  ...  9606          605066  Hs.513851   \n",
       "2  UniRef90_Q04917  UniRef50_P62258  ...  9606          113508  Hs.226755   \n",
       "3  UniRef90_P61981  UniRef50_P61981  ...  9606  605356; 617665  Hs.744840   \n",
       "4  UniRef90_P31947  UniRef50_P61981  ...  9606          601290  Hs.523718   \n",
       "\n",
       "                                                  15  \\\n",
       "0  8515476; 14702039; 11780052; 15489334; 2357255...   \n",
       "1  7644510; 8858348; 8684458; 20417184; 14702039;...   \n",
       "2  8218406; 1578511; 8561965; 8812417; 15461802; ...   \n",
       "3  10433554; 10486217; 12853948; 15489334; 235725...   \n",
       "4  1390337; 8515476; 9659898; 16710414; 15489334;...   \n",
       "\n",
       "                                                  16  \\\n",
       "0  X57346; AK292717; AL008725; CH471077; CH471077...   \n",
       "1  U20972; U54778; U43399; U43430; U28936; AB0171...   \n",
       "2  L20422; X80536; X78138; X57345; D78577; S80794...   \n",
       "3  AF142498; AB024334; CR541904; CR541925; AC0063...   \n",
       "4  M93010; X57348; AF029081; AF029082; CR541905; ...   \n",
       "\n",
       "                                                  17  \\\n",
       "0  CAA40621.1; BAF85406.1; -; EAW75893.1; EAW7589...   \n",
       "1  AAC50175.1; AAC50710.1; AAC50625.1; AAD00026.1...   \n",
       "2  AAA35483.1; CAA56676.1; CAA55017.1; CAA40620.1...   \n",
       "3  AAD48408.1; BAA85184.1; CAG46702.1; CAG46723.1...   \n",
       "4  AAA59546.1; CAA40623.1; AAC52029.1; AAC52030.1...   \n",
       "\n",
       "                                 18  \\\n",
       "0                   ENSG00000166913   \n",
       "1  ENSG00000108953; ENSG00000274474   \n",
       "2                   ENSG00000128245   \n",
       "3                   ENSG00000170027   \n",
       "4                   ENSG00000175793   \n",
       "\n",
       "                                                  19  \\\n",
       "0                   ENST00000353703; ENST00000372839   \n",
       "1  ENST00000264335; ENST00000571732; ENST00000616...   \n",
       "2                                    ENST00000248975   \n",
       "3                                    ENST00000307630   \n",
       "4                                    ENST00000339276   \n",
       "\n",
       "                                                  20  \\\n",
       "0                   ENSP00000300161; ENSP00000361930   \n",
       "1  ENSP00000264335; ENSP00000461762; ENSP00000481...   \n",
       "2                                    ENSP00000248975   \n",
       "3                                    ENSP00000306330   \n",
       "4                                    ENSP00000340989   \n",
       "\n",
       "                                                  21  \n",
       "0  11996670; 12364343; 12437930; 12468542; 124825...  \n",
       "1  15838597; 11782387; 12657644; 14966136; 153649...  \n",
       "2  10206237; 11121172; 11996670; 12176995; 124801...  \n",
       "3  11824616; 11996670; 12364343; 12482592; 150572...  \n",
       "4  10969776; 12582028; 12730237; 12787309; 145172...  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "t = pd.read_csv('../data/network/HUMAN_9606_idmapping_selected.tab', sep='\\t', header=None, low_memory=False)\n",
    "display(t.shape)\n",
    "display(t.head())\n",
    "\n",
    "entrez_to_uniprot_number = t.set_index(2)[0].to_dict()\n",
    "ensg_to_uniprot_number = t.set_index(18)[0].to_dict()\n",
    "#ensg_mapper = t[]\n",
    "#t = uni_mapper[uni_mapper[1]                                   == 'Gene_Name'].set_index(0)[2].to_dict()\n",
    "\n",
    "#next(iter(uni_mapper.items()))\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Kinase Master List (to filter)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Uniprot Protein</th>\n",
       "      <th>MS Gene</th>\n",
       "      <th>RNAseq Gene</th>\n",
       "      <th>RNAseq Accession</th>\n",
       "      <th>Family</th>\n",
       "      <th>Mouse Uniprot Protein</th>\n",
       "      <th>Mouse RNAseq gene</th>\n",
       "      <th>Mouse RNAseq Accession</th>\n",
       "      <th>Kinome Render Tree Name</th>\n",
       "      <th>Aliases (Conservative)</th>\n",
       "      <th>...</th>\n",
       "      <th>Aliases</th>\n",
       "      <th>description</th>\n",
       "      <th>other_designations</th>\n",
       "      <th>Entrez_Symbol</th>\n",
       "      <th>Old_Name</th>\n",
       "      <th>Entrez_Synonyms</th>\n",
       "      <th>Entrez_other_designations</th>\n",
       "      <th>Unnamed: 18</th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>Gene Symbol and Synonyms</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAK1</td>\n",
       "      <td>AAK1</td>\n",
       "      <td>AAK1</td>\n",
       "      <td>22848.0</td>\n",
       "      <td>Other</td>\n",
       "      <td>Aak1</td>\n",
       "      <td>Aak1</td>\n",
       "      <td>269774</td>\n",
       "      <td>AAK1</td>\n",
       "      <td>AAK1,</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AP2 associated kinase 1</td>\n",
       "      <td>adaptor-associated kinase 1</td>\n",
       "      <td>AAK1</td>\n",
       "      <td>AAK1</td>\n",
       "      <td>KIAA1048|MGC138170</td>\n",
       "      <td>adaptor-associated kinase 1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AAK1</td>\n",
       "      <td>AAK1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAPK1</td>\n",
       "      <td>PRKAA1</td>\n",
       "      <td>PRKAA1</td>\n",
       "      <td>5562.0</td>\n",
       "      <td>CAMK</td>\n",
       "      <td>Aapk1</td>\n",
       "      <td>Prkaa1</td>\n",
       "      <td>105787</td>\n",
       "      <td>AMPK[alpha]1</td>\n",
       "      <td>PRKAA1, AMPKa1</td>\n",
       "      <td>...</td>\n",
       "      <td>AMPK, AMPKa1</td>\n",
       "      <td>protein kinase, AMP-activated, alpha 1 catalyt...</td>\n",
       "      <td>5'-AMP-activated protein kinase, catalytic alp...</td>\n",
       "      <td>PRKAA1</td>\n",
       "      <td>AMPKa1</td>\n",
       "      <td>AMPK|AMPKa1|MGC33776|MGC57364</td>\n",
       "      <td>5'-AMP-activated protein kinase, catalytic alp...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PRKAA1</td>\n",
       "      <td>AMPK, PRKAA1, AMPKa1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAPK2</td>\n",
       "      <td>PRKAA2</td>\n",
       "      <td>PRKAA2</td>\n",
       "      <td>5563.0</td>\n",
       "      <td>CAMK</td>\n",
       "      <td>Aapk2</td>\n",
       "      <td>Prkaa2</td>\n",
       "      <td>108079</td>\n",
       "      <td>AMPK[alpha]2</td>\n",
       "      <td>PRKAA2, AMPK2, AMPKa2, PRKAA</td>\n",
       "      <td>...</td>\n",
       "      <td>AMPK, AMPK2, AMPKa2, PRKAA</td>\n",
       "      <td>protein kinase, AMP-activated, alpha 2 catalyt...</td>\n",
       "      <td>5'-AMP-activated protein kinase, catalytic alp...</td>\n",
       "      <td>PRKAA2</td>\n",
       "      <td>AMPKa2</td>\n",
       "      <td>AMPK|AMPK2|PRKAA</td>\n",
       "      <td>5'-AMP-activated protein kinase, catalytic alp...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PRKAA2</td>\n",
       "      <td>AMPK2, AMPK, PRKAA, AMPKa2, PRKAA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>25.0</td>\n",
       "      <td>TK</td>\n",
       "      <td>Abl1</td>\n",
       "      <td>Abl1</td>\n",
       "      <td>11350</td>\n",
       "      <td>Abl</td>\n",
       "      <td>ABL1,ABL, JTK7, bcr/abl, c-ABL, c-ABL1, v-abl</td>\n",
       "      <td>...</td>\n",
       "      <td>ABL, JTK7, bcr/abl, c-ABL, c-ABL1, p150, v-abl</td>\n",
       "      <td>ABL proto-oncogene 1, non-receptor tyrosine ki...</td>\n",
       "      <td>Abelson tyrosine-protein kinase 1|bcr/c-abl on...</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>ABL</td>\n",
       "      <td>ABL|JTK7|c-ABL|p150|v-abl</td>\n",
       "      <td>Abelson murine leukemia viral (v-abl) oncogene...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>c-ABL1, p150, ABL, JTK7, v-abl, bcr/abl, c-ABL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>ABL2</td>\n",
       "      <td>ABL2</td>\n",
       "      <td>27.0</td>\n",
       "      <td>TK</td>\n",
       "      <td>Abl2</td>\n",
       "      <td>Abl2</td>\n",
       "      <td>11352</td>\n",
       "      <td>Arg</td>\n",
       "      <td>ABL2,ABLL, ARG</td>\n",
       "      <td>...</td>\n",
       "      <td>ABLL, ARG</td>\n",
       "      <td>ABL proto-oncogene 2, non-receptor tyrosine ki...</td>\n",
       "      <td>abelson-related gene protein|c-abl oncogene 2,...</td>\n",
       "      <td>ABL2</td>\n",
       "      <td>ARG</td>\n",
       "      <td>ABLL|ARG</td>\n",
       "      <td>Abelson murine leukemia viral (v-abl) oncogene...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABL2</td>\n",
       "      <td>ABL2, ABLL, ARG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Uniprot Protein MS Gene RNAseq Gene  RNAseq Accession Family  \\\n",
       "0            AAK1    AAK1        AAK1           22848.0  Other   \n",
       "1           AAPK1  PRKAA1      PRKAA1            5562.0   CAMK   \n",
       "2           AAPK2  PRKAA2      PRKAA2            5563.0   CAMK   \n",
       "3            ABL1    ABL1        ABL1              25.0     TK   \n",
       "4            ABL2    ABL2        ABL2              27.0     TK   \n",
       "\n",
       "  Mouse Uniprot Protein Mouse RNAseq gene Mouse RNAseq Accession  \\\n",
       "0                  Aak1              Aak1                 269774   \n",
       "1                 Aapk1            Prkaa1                 105787   \n",
       "2                 Aapk2            Prkaa2                 108079   \n",
       "3                  Abl1              Abl1                  11350   \n",
       "4                  Abl2              Abl2                  11352   \n",
       "\n",
       "  Kinome Render Tree Name                         Aliases (Conservative)  ...  \\\n",
       "0                    AAK1                                          AAK1,  ...   \n",
       "1            AMPK[alpha]1                                 PRKAA1, AMPKa1  ...   \n",
       "2            AMPK[alpha]2                   PRKAA2, AMPK2, AMPKa2, PRKAA  ...   \n",
       "3                     Abl  ABL1,ABL, JTK7, bcr/abl, c-ABL, c-ABL1, v-abl  ...   \n",
       "4                     Arg                                 ABL2,ABLL, ARG  ...   \n",
       "\n",
       "                                          Aliases  \\\n",
       "0                                             NaN   \n",
       "1                                    AMPK, AMPKa1   \n",
       "2                      AMPK, AMPK2, AMPKa2, PRKAA   \n",
       "3  ABL, JTK7, bcr/abl, c-ABL, c-ABL1, p150, v-abl   \n",
       "4                                       ABLL, ARG   \n",
       "\n",
       "                                         description  \\\n",
       "0                            AP2 associated kinase 1   \n",
       "1  protein kinase, AMP-activated, alpha 1 catalyt...   \n",
       "2  protein kinase, AMP-activated, alpha 2 catalyt...   \n",
       "3  ABL proto-oncogene 1, non-receptor tyrosine ki...   \n",
       "4  ABL proto-oncogene 2, non-receptor tyrosine ki...   \n",
       "\n",
       "                                  other_designations Entrez_Symbol Old_Name  \\\n",
       "0                        adaptor-associated kinase 1          AAK1     AAK1   \n",
       "1  5'-AMP-activated protein kinase, catalytic alp...        PRKAA1   AMPKa1   \n",
       "2  5'-AMP-activated protein kinase, catalytic alp...        PRKAA2   AMPKa2   \n",
       "3  Abelson tyrosine-protein kinase 1|bcr/c-abl on...          ABL1      ABL   \n",
       "4  abelson-related gene protein|c-abl oncogene 2,...          ABL2      ARG   \n",
       "\n",
       "                 Entrez_Synonyms  \\\n",
       "0             KIAA1048|MGC138170   \n",
       "1  AMPK|AMPKa1|MGC33776|MGC57364   \n",
       "2               AMPK|AMPK2|PRKAA   \n",
       "3      ABL|JTK7|c-ABL|p150|v-abl   \n",
       "4                       ABLL|ARG   \n",
       "\n",
       "                           Entrez_other_designations Unnamed: 18  Gene Symbol  \\\n",
       "0                        adaptor-associated kinase 1         NaN         AAK1   \n",
       "1  5'-AMP-activated protein kinase, catalytic alp...         NaN       PRKAA1   \n",
       "2  5'-AMP-activated protein kinase, catalytic alp...         NaN       PRKAA2   \n",
       "3  Abelson murine leukemia viral (v-abl) oncogene...         NaN         ABL1   \n",
       "4  Abelson murine leukemia viral (v-abl) oncogene...         NaN         ABL2   \n",
       "\n",
       "                            Gene Symbol and Synonyms  \n",
       "0                                               AAK1  \n",
       "1                               AMPK, PRKAA1, AMPKa1  \n",
       "2                 AMPK2, AMPK, PRKAA, AMPKa2, PRKAA2  \n",
       "3  c-ABL1, p150, ABL, JTK7, v-abl, bcr/abl, c-ABL...  \n",
       "4                                    ABL2, ABLL, ARG  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kmast = pd.read_excel('../data/KINASESmasterlist_w_Aliases.xlsx')\n",
    "kmast.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['SGK', 'SGK2', 'SGKL', 'C8orf44-SGK3', 'SGK1', 'SGK3', 'SGK196']"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[x for x,y in hgnc_mapper.items() if 'SGK' in y]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'MAPK3'"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_mapper['MAPK3']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'AAK1': 'AAK1',\n",
       " 'PRKAA1': 'AAPK1',\n",
       " 'PRKAA2': 'AAPK2',\n",
       " 'ABL1': 'ABL1',\n",
       " 'ABL2': 'ABL2',\n",
       " 'TNK2': 'ACK1',\n",
       " 'ACVR1B': 'ACV1B',\n",
       " 'ACVR1C': 'ACV1C',\n",
       " 'ACVRL1': 'ACVL1',\n",
       " 'ACVR1': 'ACVR1',\n",
       " 'ADCK1': 'ADCK1',\n",
       " 'ADCK2': 'ADCK2',\n",
       " 'ADCK3': 'ADCK3',\n",
       " 'ADCK4': 'ADCK4',\n",
       " 'ADCK5': 'ADCK5',\n",
       " 'ADK': 'ADK',\n",
       " 'AGK': 'AGK',\n",
       " 'AKT1': 'AKT1',\n",
       " 'AKT2': 'AKT2',\n",
       " 'AKT3': 'AKT3',\n",
       " 'ALK': 'ALK',\n",
       " 'ALPK1': 'ALPK1',\n",
       " 'ALPK2': 'ALPK2',\n",
       " 'ALPK3': 'ALPK3',\n",
       " 'AMHR2': 'AMHR2',\n",
       " 'ANKK1': 'ANKK1',\n",
       " 'NPR1': 'ANPRA',\n",
       " 'NPR2': 'ANPRB',\n",
       " 'ARAF': 'ARAF',\n",
       " 'ADRBK1': 'ARBK1',\n",
       " 'ADRBK2': 'ARBK2',\n",
       " 'ATM': 'ATM',\n",
       " 'ATR': 'ATR',\n",
       " 'AURKB': 'AURKB',\n",
       " 'AURKC': 'AURKC',\n",
       " 'ACVR2A': 'AVR2A',\n",
       " 'ACVR2B': 'AVR2B',\n",
       " 'BCKDK': 'BCKD',\n",
       " 'BLK': 'BLK',\n",
       " 'BMP2K': 'BMP2K',\n",
       " 'BMPR2': 'BMPR2',\n",
       " 'BMPR1A': 'BMR1A',\n",
       " 'BMPR1B': 'BMR1B',\n",
       " 'BMX': 'BMX',\n",
       " 'BRAF': 'BRAF',\n",
       " 'BRD2': 'BRD2',\n",
       " 'BRD3': 'BRD3',\n",
       " 'BRD4': 'BRD4',\n",
       " 'BRSK1': 'BRSK1',\n",
       " 'BRSK2': 'BRSK2',\n",
       " 'BTK': 'BTK',\n",
       " 'BUB1': 'BUB1',\n",
       " 'BUB1B': 'BUB1B',\n",
       " 'CAMKV': 'CAMKV',\n",
       " 'CDK11A': 'CD11A',\n",
       " 'CDK11B': 'CD11B',\n",
       " 'CDC7': 'CDC7',\n",
       " 'CDK1': 'CDK1',\n",
       " 'CDK10': 'CDK10',\n",
       " 'CDK12': 'CDK12',\n",
       " 'CDK13': 'CDK13',\n",
       " 'CDK14': 'CDK14',\n",
       " 'CDK15': 'CDK15',\n",
       " 'CDK16': 'CDK16',\n",
       " 'CDK17': 'CDK17',\n",
       " 'CDK18': 'CDK18',\n",
       " 'CDK19': 'CDK19',\n",
       " 'CDK2': 'CDK2',\n",
       " 'CDK20': 'CDK20',\n",
       " 'CDK3': 'CDK3',\n",
       " 'CDK4': 'CDK4',\n",
       " 'CDK5': 'CDK5',\n",
       " 'CDK6': 'CDK6',\n",
       " 'CDK7': 'CDK7',\n",
       " 'CDK8': 'CDK8',\n",
       " 'CDK9': 'CDK9',\n",
       " 'CDKL1': 'CDKL1',\n",
       " 'CDKL2': 'CDKL2',\n",
       " 'CDKL3': 'CDKL3',\n",
       " 'CDKL4': 'CDKL4',\n",
       " 'CDKL5': 'CDKL5',\n",
       " 'CHEK1': 'CHK1',\n",
       " 'CHEK2': 'CHK2',\n",
       " 'CHKA': 'CHKA',\n",
       " 'CHKB': 'CHKB',\n",
       " 'CLK1': 'CLK1',\n",
       " 'CLK2': 'CLK2',\n",
       " 'CLK3': 'CLK3',\n",
       " 'CLK4': 'CLK4',\n",
       " 'CSF1R': 'CSF1R',\n",
       " 'CSK': 'CSK',\n",
       " 'CSNK2A1': 'CSK21',\n",
       " 'CSNK2A2': 'CSK22',\n",
       " 'CSNK2A3': 'CSK23',\n",
       " 'CASK': 'CSKP',\n",
       " 'CIT': 'CTRO',\n",
       " 'DAPK1': 'DAPK1',\n",
       " 'DAPK2': 'DAPK2',\n",
       " 'DAPK3': 'DAPK3',\n",
       " 'DCK': 'DCK',\n",
       " 'DCLK1': 'DCLK1',\n",
       " 'DCLK2': 'DCLK2',\n",
       " 'DCLK3': 'DCLK3',\n",
       " 'DDR1': 'DDR1',\n",
       " 'DDR2': 'DDR2',\n",
       " 'DGKQ': 'DGKQ',\n",
       " 'DMPK': 'DMPK',\n",
       " 'DSTYK': 'DUSTY',\n",
       " 'DYRK1A': 'DYR1A',\n",
       " 'DYRK1B': 'DYR1B',\n",
       " 'DYRK2': 'DYRK2',\n",
       " 'DYRK3': 'DYRK3',\n",
       " 'DYRK4': 'DYRK4',\n",
       " 'EIF2AK1': 'E2AK1',\n",
       " 'EIF2AK2': 'E2AK2',\n",
       " 'EIF2AK3': 'E2AK3',\n",
       " 'EIF2AK4': 'E2AK4',\n",
       " 'EEF2K': 'EF2K',\n",
       " 'EGFR': 'EGFR',\n",
       " 'EPHA1': 'EPHA1',\n",
       " 'EPHA2': 'EPHA2',\n",
       " 'EPHA3': 'EPHA3',\n",
       " 'EPHA4': 'EPHA4',\n",
       " 'EPHA5': 'EPHA5',\n",
       " 'EPHA6': 'EPHA6',\n",
       " 'EPHA7': 'EPHA7',\n",
       " 'EPHA8': 'EPHA8',\n",
       " 'EPHA10': 'EPHAA',\n",
       " 'EPHB1': 'EPHB1',\n",
       " 'EPHB2': 'EPHB2',\n",
       " 'EPHB3': 'EPHB3',\n",
       " 'EPHB4': 'EPHB4',\n",
       " 'EPHB6': 'EPHB6',\n",
       " 'ERBB2': 'ERBB2',\n",
       " 'ERBB3': 'ERBB3',\n",
       " 'ERBB4': 'ERBB4',\n",
       " 'ERN1': 'ERN1',\n",
       " 'ERN2': 'ERN2',\n",
       " 'PTK2': 'FAK1',\n",
       " 'PTK2B': 'FAK2',\n",
       " 'FASTKD5': 'FAKD5',\n",
       " 'FASTK': 'FASTK',\n",
       " 'FER': 'FER',\n",
       " 'FES': 'FES',\n",
       " 'FGFR1': 'FGFR1',\n",
       " 'FGFR2': 'FGFR2',\n",
       " 'FGFR3': 'FGFR3',\n",
       " 'FGFR4': 'FGFR4',\n",
       " 'FGR': 'FGR',\n",
       " 'FLT3': 'FLT3',\n",
       " 'FN3K': 'FN3K',\n",
       " 'FRK': 'FRK',\n",
       " 'FUK': 'FUK',\n",
       " 'FYN': 'FYN',\n",
       " 'PIKFYVE': 'FYV1',\n",
       " 'GAK': 'GAK',\n",
       " 'GALK1': 'GALK1',\n",
       " 'GYK': 'GLPK',\n",
       " 'GRK4': 'GRK4',\n",
       " 'GRK5': 'GRK5',\n",
       " 'GRK6': 'GRK6',\n",
       " 'GRK7': 'GRK7',\n",
       " 'GSK3A': 'GSK3A',\n",
       " 'GSK3B': 'GSK3B',\n",
       " 'GUCY2C': 'GUC2C',\n",
       " 'GUCY2D': 'GUC2D',\n",
       " 'GUCY2F': 'GUC2F',\n",
       " 'MASTL': 'GWL',\n",
       " 'GSG2': 'HASP',\n",
       " 'HCK': 'HCK',\n",
       " 'HIPK1': 'HIPK1',\n",
       " 'HIPK2': 'HIPK2',\n",
       " 'HIPK3': 'HIPK3',\n",
       " 'HIPK4': 'HIPK4',\n",
       " 'HUNK': 'HUNK',\n",
       " 'HK1': 'HXK1',\n",
       " 'HK2': 'HXK2',\n",
       " 'HYKK': 'HYKK',\n",
       " 'ICK': 'ICK',\n",
       " 'IGF1R': 'IGF1R',\n",
       " 'CHUK': 'IKKA',\n",
       " 'IKBKB': 'IKKB',\n",
       " 'IKBKE': 'IKKE',\n",
       " 'ILK': 'ILK',\n",
       " 'INSR': 'INSR',\n",
       " 'INSRR': 'INSRR',\n",
       " 'IP6K1': 'IP6K1',\n",
       " 'IP6K2': 'IP6K2',\n",
       " 'IP6K3': 'IP6K3',\n",
       " 'IPMK': 'IPMK',\n",
       " 'IRAK1': 'IRAK1',\n",
       " 'IRAK2': 'IRAK2',\n",
       " 'IRAK3': 'IRAK3',\n",
       " 'IRAK4': 'IRAK4',\n",
       " 'ITK': 'ITK',\n",
       " 'JAK1': 'JAK1',\n",
       " 'JAK2': 'JAK2',\n",
       " 'JAK3': 'JAK3',\n",
       " 'PFKM': 'K6PF',\n",
       " 'PFKL': 'K6PL',\n",
       " 'PFKP': 'K6PP',\n",
       " 'AK1': 'KAD1',\n",
       " 'AK2': 'KAD2',\n",
       " 'KALRN': 'KALRN',\n",
       " 'PRKACA': 'KAPCA',\n",
       " 'PRKACB': 'KAPCB',\n",
       " 'PRKACG': 'KAPCG',\n",
       " 'CSNK1A1': 'KC1A',\n",
       " 'CSNK1A1L': 'KC1AL',\n",
       " 'CSNK1D': 'KC1D',\n",
       " 'CSNK1E': 'KC1E',\n",
       " 'CSNK1G1': 'KC1G1',\n",
       " 'CSNK1G2': 'KC1G2',\n",
       " 'CSNK1G3': 'KC1G3',\n",
       " 'CAMK1': 'KCC1A',\n",
       " 'PNCK': 'KCC1B',\n",
       " 'CAMK1D': 'KCC1D',\n",
       " 'CAMK1G': 'KCC1G',\n",
       " 'CAMK2A': 'KCC2A',\n",
       " 'CAMK2B': 'KCC2B',\n",
       " 'CAMK2D': 'KCC2D',\n",
       " 'CAMK2G': 'KCC2G',\n",
       " 'CAMK4': 'KCC4',\n",
       " 'CKM': 'KCRM',\n",
       " 'CKMT2': 'KCRS',\n",
       " 'PRKG1': 'KGP1',\n",
       " 'PRKG2': 'KGP2',\n",
       " 'KHK': 'KHK',\n",
       " 'KIT': 'KIT',\n",
       " 'TK2': 'KITM',\n",
       " 'CAMKK1': 'KKCC1',\n",
       " 'CAMKK2': 'KKCC2',\n",
       " 'PRKCA': 'KPCA',\n",
       " 'PRKCB': 'KPCB',\n",
       " 'PRKCD': 'KPCD',\n",
       " 'PRKD1': 'KPCD1',\n",
       " 'PRKD2': 'KPCD2',\n",
       " 'PRKD3': 'KPCD3',\n",
       " 'PRKCE': 'KPCE',\n",
       " 'PRKCG': 'KPCG',\n",
       " 'PRKCI': 'KPCI',\n",
       " 'PRKCH': 'KPCL',\n",
       " 'PRKCQ': 'KPCT',\n",
       " 'PRKCZ': 'KPCZ',\n",
       " 'PSKH1': 'KPSH1',\n",
       " 'PSKH2': 'KPSH2',\n",
       " 'PKM': 'KPYM',\n",
       " 'RPS6KA1': 'KS6A1',\n",
       " 'RPS6KA2': 'KS6A2',\n",
       " 'RPS6KA3': 'KS6A3',\n",
       " 'RPS6KA4': 'KS6A4',\n",
       " 'RPS6KA5': 'KS6A5',\n",
       " 'RPS6KA6': 'KS6A6',\n",
       " 'RPS6KB1': 'KS6B1',\n",
       " 'RPS6KB2': 'KS6B2',\n",
       " 'RPS6KC1': 'KS6C1',\n",
       " 'KSR1': 'KSR1',\n",
       " 'KSR2': 'KSR2',\n",
       " 'SYK': 'KSYK',\n",
       " 'FN3KRP': 'KT3K',\n",
       " 'LATS1': 'LATS1',\n",
       " 'LATS2': 'LATS2',\n",
       " 'LCK': 'LCK',\n",
       " 'LIMK1': 'LIMK1',\n",
       " 'LIMK2': 'LIMK2',\n",
       " 'AATK': 'LMTK1',\n",
       " 'LMTK2': 'LMTK2',\n",
       " 'LMTK3': 'LMTK3',\n",
       " 'LRRK1': 'LRRK1',\n",
       " 'LRRK2': 'LRRK2',\n",
       " 'LTK': 'LTK',\n",
       " 'LYN': 'LYN',\n",
       " 'MAP3K1': 'M3K1',\n",
       " 'MAP3K10': 'M3K10',\n",
       " 'MAP3K11': 'M3K11',\n",
       " 'MAP3K12': 'M3K12',\n",
       " 'MAP3K13': 'M3K13',\n",
       " 'MAP3K14': 'M3K14',\n",
       " 'MAP3K15': 'M3K15',\n",
       " 'MAP3K19': 'M3K19',\n",
       " 'MAP3K2': 'M3K2',\n",
       " 'MAP3K3': 'M3K3',\n",
       " 'MAP3K4': 'M3K4',\n",
       " 'MAP3K5': 'M3K5',\n",
       " 'MAP3K6': 'M3K6',\n",
       " 'MAP3K7': 'M3K7',\n",
       " 'MAP3K8': 'M3K8',\n",
       " 'MAP3K9': 'M3K9',\n",
       " 'MLK4': 'M3KL4',\n",
       " 'MAP4K1': 'M4K1',\n",
       " 'MAP4K2': 'M4K2',\n",
       " 'MAP4K3': 'M4K3',\n",
       " 'MAP4K4': 'M4K4',\n",
       " 'MAP4K5': 'M4K5',\n",
       " 'MAK': 'MAK',\n",
       " 'MAPKAPK2': 'MAPK2',\n",
       " 'MAPKAPK3': 'MAPK3',\n",
       " 'MAPKAPK5': 'MAPK5',\n",
       " 'MARK1': 'MARK1',\n",
       " 'MARK2': 'MARK2',\n",
       " 'MARK3': 'MARK3',\n",
       " 'MARK4': 'MARK4',\n",
       " 'MAST1': 'MAST1',\n",
       " 'MAST2': 'MAST2',\n",
       " 'MAST3': 'MAST3',\n",
       " 'MAST4': 'MAST4',\n",
       " 'MATK': 'MATK',\n",
       " 'MELK': 'MELK',\n",
       " 'MERTK': 'MERTK',\n",
       " 'MET': 'MET',\n",
       " 'MINK1': 'MINK1',\n",
       " 'MAPK1': 'MK01',\n",
       " 'MAPK3': 'MK03',\n",
       " 'MAPK4': 'MK04',\n",
       " 'MAPK6': 'MK06',\n",
       " 'MAPK7': 'MK07',\n",
       " 'MAPK8': 'MK08',\n",
       " 'MAPK9': 'MK09',\n",
       " 'MAPK10': 'MK10',\n",
       " 'MAPK11': 'MK11',\n",
       " 'MAPK12': 'MK12',\n",
       " 'MAPK13': 'MK13',\n",
       " 'MAPK14': 'MK14',\n",
       " 'MAPK15': 'MK15',\n",
       " 'MKNK1': 'MKNK1',\n",
       " 'MKNK2': 'MKNK2',\n",
       " 'MLKL': 'MLKL',\n",
       " 'MLTK': 'MLTK',\n",
       " 'MOK': 'MOK',\n",
       " 'MOS': 'MOS',\n",
       " 'MAP2K1': 'MP2K1',\n",
       " 'MAP2K2': 'MP2K2',\n",
       " 'MAP2K3': 'MP2K3',\n",
       " 'MAP2K4': 'MP2K4',\n",
       " 'MAP2K5': 'MP2K5',\n",
       " 'MAP2K6': 'MP2K6',\n",
       " 'MAP2K7': 'MP2K7',\n",
       " 'CDC42BPA': 'MRCKA',\n",
       " 'CDC42BPB': 'MRCKB',\n",
       " 'CDC42BPG': 'MRCKG',\n",
       " 'MST4': 'MST4',\n",
       " 'MTOR': 'MTOR',\n",
       " 'MUSK': 'MUSK',\n",
       " 'MYLK': 'MYLK',\n",
       " 'MYLK2': 'MYLK2',\n",
       " 'MYLK3': 'MYLK3',\n",
       " 'MYLK4': 'MYLK4',\n",
       " 'MYO3A': 'MYO3A',\n",
       " 'MYO3B': 'MYO3B',\n",
       " 'NME3': 'NDK3',\n",
       " 'NME5': 'NDK5',\n",
       " 'NME6': 'NDK6',\n",
       " 'NME7': 'NDK7',\n",
       " 'NME1': 'NDKA',\n",
       " 'NME2': 'NDKB',\n",
       " 'NME4': 'NDKM',\n",
       " 'NEK1': 'NEK1',\n",
       " 'NEK10': 'NEK10',\n",
       " 'NEK11': 'NEK11',\n",
       " 'NEK2': 'NEK2',\n",
       " 'NEK3': 'NEK3',\n",
       " 'NEK4': 'NEK4',\n",
       " 'NEK5': 'NEK5',\n",
       " 'NEK6': 'NEK6',\n",
       " 'NEK7': 'NEK7',\n",
       " 'NEK8': 'NEK8',\n",
       " 'NEK9': 'NEK9',\n",
       " 'NIM1': 'NIM1',\n",
       " 'NLK': 'NLK',\n",
       " 'NRBP1': 'NRBP',\n",
       " 'NRBP2': 'NRBP2',\n",
       " 'NRK': 'NRK',\n",
       " 'SCYL1': 'NTKL',\n",
       " 'NTRK1': 'NTRK1',\n",
       " 'NTRK2': 'NTRK2',\n",
       " 'NTRK3': 'NTRK3',\n",
       " 'NUAK1': 'NUAK1',\n",
       " 'NUAK2': 'NUAK2',\n",
       " 'OBSCN': 'OBSCN',\n",
       " 'OXSR1': 'OXSR1',\n",
       " 'PIK3C2A': 'P3C2A',\n",
       " 'PI4K2B': 'P4K2B',\n",
       " 'SCYL3': 'PACE1',\n",
       " 'PAK1': 'PAK1',\n",
       " 'PAK2': 'PAK2',\n",
       " 'PAK3': 'PAK3',\n",
       " 'PAK4': 'PAK4',\n",
       " 'PAK6': 'PAK6',\n",
       " 'PAK7': 'PAK7',\n",
       " 'PAN3': 'PAN3',\n",
       " 'PASK': 'PASK',\n",
       " 'PCK2': 'PCKGM',\n",
       " 'PDK1': 'PDK1',\n",
       " 'PDIK1L': 'PDK1L',\n",
       " 'PDK2': 'PDK2',\n",
       " 'PDK3': 'PDK3',\n",
       " 'PDK4': 'PDK4',\n",
       " 'PDPK1': 'PDPK1',\n",
       " 'PDPK2': 'PDPK2',\n",
       " 'PDXK': 'PDXK',\n",
       " 'PEAK1': 'PEAK1',\n",
       " 'PDGFRA': 'PGFRA',\n",
       " 'PDGFRB': 'PGFRB',\n",
       " 'PGK1': 'PGK1',\n",
       " 'PGK2': 'PGK2',\n",
       " 'PHKG1': 'PHKG1',\n",
       " 'PHKG2': 'PHKG2',\n",
       " 'PIK3R4': 'PI3R4',\n",
       " 'PIP4K2A': 'PI42A',\n",
       " 'PIP4K2B': 'PI42B',\n",
       " 'PIP4K2C': 'PI42C',\n",
       " 'PI4KA': 'PI4KA',\n",
       " 'PIP5K1A': 'PI51A',\n",
       " 'PIP5K1B': 'PI51B',\n",
       " 'PIP5K1C': 'PI51C',\n",
       " 'PIM1': 'PIM1',\n",
       " 'PIM2': 'PIM2',\n",
       " 'PIM3': 'PIM3',\n",
       " 'PINK1': 'PINK1',\n",
       " 'PIK3C3': 'PK3C3',\n",
       " 'PIK3CA': 'PK3CA',\n",
       " 'PIK3CB': 'PK3CB',\n",
       " 'PIK3CD': 'PK3CD',\n",
       " 'PIK3CG': 'PK3CG',\n",
       " 'PKDCC': 'PKDCC',\n",
       " 'PKN1': 'PKN1',\n",
       " 'PKN2': 'PKN2',\n",
       " 'PKN3': 'PKN3',\n",
       " 'PLK1': 'PLK1',\n",
       " 'PLK2': 'PLK2',\n",
       " 'PLK3': 'PLK3',\n",
       " 'PLK4': 'PLK4',\n",
       " 'PLK5': 'PLK5',\n",
       " 'PKMYT1': 'PMYT1',\n",
       " 'PRKDC': 'PRKDC',\n",
       " 'PRKX': 'PRKX',\n",
       " 'PRKY': 'PRKY',\n",
       " 'PRPF4B': 'PRP4B',\n",
       " 'TP53RK': 'PRPK',\n",
       " 'PTK6': 'PTK6',\n",
       " 'PTK7': 'PTK7',\n",
       " 'PXK': 'PXK',\n",
       " 'RAF1': 'RAF1',\n",
       " 'RET': 'RET',\n",
       " 'RIOK1': 'RIOK1',\n",
       " 'RIOK2': 'RIOK2',\n",
       " 'RIOK3': 'RIOK3',\n",
       " 'RIPK1': 'RIPK1',\n",
       " 'RIPK2': 'RIPK2',\n",
       " 'RIPK3': 'RIPK3',\n",
       " 'RIPK4': 'RIPK4',\n",
       " 'GRK1': 'RK',\n",
       " 'RNASEL': 'RN5A',\n",
       " 'ROCK1': 'ROCK1',\n",
       " 'ROCK2': 'ROCK2',\n",
       " 'MST1R': 'RON',\n",
       " 'ROR1': 'ROR1',\n",
       " 'ROR2': 'ROR2',\n",
       " 'ROS1': 'ROS1',\n",
       " 'RPS6KL1': 'RPKL1',\n",
       " 'RYK': 'RYK',\n",
       " 'SBK1': 'SBK1',\n",
       " 'SBK2': 'SBK2',\n",
       " 'SCYL2': 'SCYL2',\n",
       " 'SGK110': 'SG110',\n",
       " 'SGK196': 'SG196',\n",
       " 'SGK223': 'SG223',\n",
       " 'SGK494': 'SG494',\n",
       " 'SGK1': 'SGK1',\n",
       " 'SGK2': 'SGK2',\n",
       " 'SGK3': 'SGK3',\n",
       " 'SGK071': 'SGK71',\n",
       " 'SIK1': 'SIK1',\n",
       " 'SIK2': 'SIK2',\n",
       " 'SIK3': 'SIK3',\n",
       " 'SLK': 'SLK',\n",
       " 'SMG1': 'SMG1',\n",
       " 'SNRK': 'SNRK',\n",
       " 'SPEG': 'SPEG',\n",
       " 'SRC': 'SRC',\n",
       " 'SRMS': 'SRMS',\n",
       " 'SRPK1': 'SRPK1',\n",
       " 'SRPK2': 'SRPK2',\n",
       " 'SRPK3': 'SRPK3',\n",
       " 'STK17A': 'ST17A',\n",
       " 'STK17B': 'ST17B',\n",
       " 'STK32A': 'ST32A',\n",
       " 'STK32B': 'ST32B',\n",
       " 'STK32C': 'ST32C',\n",
       " 'STK38L': 'ST38L',\n",
       " 'STK10': 'STK10',\n",
       " 'STK11': 'STK11',\n",
       " 'STK16': 'STK16',\n",
       " 'STK19': 'STK19',\n",
       " 'STK24': 'STK24',\n",
       " 'STK25': 'STK25',\n",
       " 'STK3': 'STK3',\n",
       " 'STK31': 'STK31',\n",
       " 'STK33': 'STK33',\n",
       " 'STK35': 'STK35',\n",
       " 'STK36': 'STK36',\n",
       " 'STK38': 'STK38',\n",
       " 'STK39': 'STK39',\n",
       " 'STK4': 'STK4',\n",
       " 'STK40': 'STK40',\n",
       " 'AURKA': 'STK6',\n",
       " 'STRADA': 'STRAA',\n",
       " 'STRADB': 'STRAB',\n",
       " 'STYK1': 'STYK1',\n",
       " 'TAOK1': 'TAOK1',\n",
       " 'TAOK2': 'TAOK2',\n",
       " 'TAOK3': 'TAOK3',\n",
       " 'TBCK': 'TBCK',\n",
       " 'TBK1': 'TBK1',\n",
       " 'TEC': 'TEC',\n",
       " 'TESK1': 'TESK1',\n",
       " 'TESK2': 'TESK2',\n",
       " 'TEX14': 'TEX14',\n",
       " 'TGFBR1': 'TGFR1',\n",
       " 'TGFBR2': 'TGFR2',\n",
       " 'TIE1': 'TIE1',\n",
       " 'TEK': 'TIE2',\n",
       " 'TRIM24': 'TIF1A',\n",
       " 'TRIM28': 'TIF1B',\n",
       " 'TTN': 'TITIN',\n",
       " 'TLK1': 'TLK1',\n",
       " 'TLK2': 'TLK2',\n",
       " 'TNNI3K': 'TNI3K',\n",
       " 'TNIK': 'TNIK',\n",
       " 'TNK1': 'TNK1',\n",
       " 'PBK': 'TOPK',\n",
       " 'TRIB1': 'TRIB1',\n",
       " 'TRIB2': 'TRIB2',\n",
       " 'TRIB3': 'TRIB3',\n",
       " 'TRIO': 'TRIO',\n",
       " 'TRPM6': 'TRPM6',\n",
       " 'TRPM7': 'TRPM7',\n",
       " 'TSSK1B': 'TSSK1',\n",
       " 'TSSK2': 'TSSK2',\n",
       " 'TSSK3': 'TSSK3',\n",
       " 'TSSK4': 'TSSK4',\n",
       " 'TSSK6': 'TSSK6',\n",
       " 'TTBK1': 'TTBK1',\n",
       " 'TTBK2': 'TTBK2',\n",
       " 'TTK': 'TTK',\n",
       " 'TXK': 'TXK',\n",
       " 'TYK2': 'TYK2',\n",
       " 'TYRO3': 'TYRO3',\n",
       " 'AXL': 'UFO',\n",
       " 'UHMK1': 'UHMK1',\n",
       " 'ULK1': 'ULK1',\n",
       " 'ULK2': 'ULK2',\n",
       " 'ULK3': 'ULK3',\n",
       " 'ULK4': 'ULK4',\n",
       " 'FLT1': 'VGFR1',\n",
       " 'KDR': 'VGFR2',\n",
       " 'FLT4': 'VGFR3',\n",
       " 'PPIP5K1': 'VIP1',\n",
       " 'PPIP5K2': 'VIP2',\n",
       " 'VRK1': 'VRK1',\n",
       " 'VRK2': 'VRK2',\n",
       " 'VRK3': 'VRK3',\n",
       " 'WEE1': 'WEE1',\n",
       " 'WEE2': 'WEE2',\n",
       " 'WNK1': 'WNK1',\n",
       " 'WNK2': 'WNK2',\n",
       " 'WNK3': 'WNK3',\n",
       " 'WNK4': 'WNK4',\n",
       " 'YES1': 'YES',\n",
       " 'ZAP70': 'ZAP70'}"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dict(zip(kmast['MS Gene'], kmast['Uniprot Protein']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "567"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hard_coded = {'ACV1B':'ACVR1B', 'PDPK2':'PDK2', 'SGK110':'SBK3', 'SGK223':'PRAG1'}\n",
    "hgnc_mapper.update(hard_coded)\n",
    "kinome_set = set(hgnc_mapper[x] for x in kmast['MS Gene'] if 'GYK' != x and 'SGK494' != x)# if x in hgnc_mapper.keys())\n",
    "\n",
    "\n",
    "uniprot_to_hgnc_map_helper = {y:hgnc_mapper[x] for x,y in zip(kmast['MS Gene'], kmast['Uniprot Protein']) if 'GYK' != x and 'SGK494' != x}\n",
    "uniprot_to_hgnc_map_helper.update({'RAC1':'RAC1'})\n",
    "#hgnc_mapper.update(uniprot_to_hgnc_map_helper)\n",
    "# ^ this causes a lot of collision issues e.g. mapk3 and mapkapk3\n",
    "\n",
    "len(kinome_set)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Filter our uniprot queries down"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "570\n",
      "499\n",
      "744\n"
     ]
    }
   ],
   "source": [
    "print(len(set(kmast['Uniprot Protein'])))\n",
    "print(len(set(kmast['Symbol'])))\n",
    "print(len(set(kmast['Symbol'])|set(kmast['Uniprot Protein'])))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'AAK1',\n",
       " 'AAPK1',\n",
       " 'AAPK2',\n",
       " 'ABL1',\n",
       " 'ABL2',\n",
       " 'ACK1',\n",
       " 'ACV1B',\n",
       " 'ACV1C',\n",
       " 'ACVL1',\n",
       " 'ACVR1',\n",
       " 'ADCK1',\n",
       " 'ADCK2',\n",
       " 'ADCK3',\n",
       " 'ADCK4',\n",
       " 'ADCK5',\n",
       " 'ADK',\n",
       " 'AGK',\n",
       " 'AKT1',\n",
       " 'AKT2',\n",
       " 'AKT3',\n",
       " 'ALK',\n",
       " 'ALPK1',\n",
       " 'ALPK2',\n",
       " 'ALPK3',\n",
       " 'AMHR2',\n",
       " 'ANKK1',\n",
       " 'ANPRA',\n",
       " 'ANPRB',\n",
       " 'ARAF',\n",
       " 'ARBK1',\n",
       " 'ARBK2',\n",
       " 'ATM',\n",
       " 'ATR',\n",
       " 'AURKB',\n",
       " 'AURKC',\n",
       " 'AVR2A',\n",
       " 'AVR2B',\n",
       " 'BCKD',\n",
       " 'BLK',\n",
       " 'BMP2K',\n",
       " 'BMPR2',\n",
       " 'BMR1A',\n",
       " 'BMR1B',\n",
       " 'BMX',\n",
       " 'BRAF',\n",
       " 'BRD2',\n",
       " 'BRD3',\n",
       " 'BRD4',\n",
       " 'BRSK1',\n",
       " 'BRSK2',\n",
       " 'BTK',\n",
       " 'BUB1',\n",
       " 'BUB1B',\n",
       " 'CAMKV',\n",
       " 'CD11A',\n",
       " 'CD11B',\n",
       " 'CDC7',\n",
       " 'CDK1',\n",
       " 'CDK10',\n",
       " 'CDK12',\n",
       " 'CDK13',\n",
       " 'CDK14',\n",
       " 'CDK15',\n",
       " 'CDK16',\n",
       " 'CDK17',\n",
       " 'CDK18',\n",
       " 'CDK19',\n",
       " 'CDK2',\n",
       " 'CDK20',\n",
       " 'CDK3',\n",
       " 'CDK4',\n",
       " 'CDK5',\n",
       " 'CDK6',\n",
       " 'CDK7',\n",
       " 'CDK8',\n",
       " 'CDK9',\n",
       " 'CDKL1',\n",
       " 'CDKL2',\n",
       " 'CDKL3',\n",
       " 'CDKL4',\n",
       " 'CDKL5',\n",
       " 'CHK1',\n",
       " 'CHK2',\n",
       " 'CHKA',\n",
       " 'CHKB',\n",
       " 'CLK1',\n",
       " 'CLK2',\n",
       " 'CLK3',\n",
       " 'CLK4',\n",
       " 'CSF1R',\n",
       " 'CSK',\n",
       " 'CSK21',\n",
       " 'CSK22',\n",
       " 'CSK23',\n",
       " 'CSKP',\n",
       " 'CTRO',\n",
       " 'DAPK1',\n",
       " 'DAPK2',\n",
       " 'DAPK3',\n",
       " 'DCK',\n",
       " 'DCLK1',\n",
       " 'DCLK2',\n",
       " 'DCLK3',\n",
       " 'DDR1',\n",
       " 'DDR2',\n",
       " 'DGKQ',\n",
       " 'DMPK',\n",
       " 'DUSTY',\n",
       " 'DYR1A',\n",
       " 'DYR1B',\n",
       " 'DYRK2',\n",
       " 'DYRK3',\n",
       " 'DYRK4',\n",
       " 'E2AK1',\n",
       " 'E2AK2',\n",
       " 'E2AK3',\n",
       " 'E2AK4',\n",
       " 'EF2K',\n",
       " 'EGFR',\n",
       " 'EPHA1',\n",
       " 'EPHA2',\n",
       " 'EPHA3',\n",
       " 'EPHA4',\n",
       " 'EPHA5',\n",
       " 'EPHA6',\n",
       " 'EPHA7',\n",
       " 'EPHA8',\n",
       " 'EPHAA',\n",
       " 'EPHB1',\n",
       " 'EPHB2',\n",
       " 'EPHB3',\n",
       " 'EPHB4',\n",
       " 'EPHB6',\n",
       " 'ERBB2',\n",
       " 'ERBB3',\n",
       " 'ERBB4',\n",
       " 'ERN1',\n",
       " 'ERN2',\n",
       " 'FAK1',\n",
       " 'FAK2',\n",
       " 'FAKD5',\n",
       " 'FASTK',\n",
       " 'FER',\n",
       " 'FES',\n",
       " 'FGFR1',\n",
       " 'FGFR2',\n",
       " 'FGFR3',\n",
       " 'FGFR4',\n",
       " 'FGR',\n",
       " 'FLT3',\n",
       " 'FN3K',\n",
       " 'FRK',\n",
       " 'FUK',\n",
       " 'FYN',\n",
       " 'FYV1',\n",
       " 'GAK',\n",
       " 'GALK1',\n",
       " 'GLPK',\n",
       " 'GRK4',\n",
       " 'GRK5',\n",
       " 'GRK6',\n",
       " 'GRK7',\n",
       " 'GSK3A',\n",
       " 'GSK3B',\n",
       " 'GUC2C',\n",
       " 'GUC2D',\n",
       " 'GUC2F',\n",
       " 'GWL',\n",
       " 'HASP',\n",
       " 'HCK',\n",
       " 'HIPK1',\n",
       " 'HIPK2',\n",
       " 'HIPK3',\n",
       " 'HIPK4',\n",
       " 'HUNK',\n",
       " 'HXK1',\n",
       " 'HXK2',\n",
       " 'HYKK',\n",
       " 'ICK',\n",
       " 'IGF1R',\n",
       " 'IKKA',\n",
       " 'IKKB',\n",
       " 'IKKE',\n",
       " 'ILK',\n",
       " 'INSR',\n",
       " 'INSRR',\n",
       " 'IP6K1',\n",
       " 'IP6K2',\n",
       " 'IP6K3',\n",
       " 'IPMK',\n",
       " 'IRAK1',\n",
       " 'IRAK2',\n",
       " 'IRAK3',\n",
       " 'IRAK4',\n",
       " 'ITK',\n",
       " 'JAK1',\n",
       " 'JAK2',\n",
       " 'JAK3',\n",
       " 'K6PF',\n",
       " 'K6PL',\n",
       " 'K6PP',\n",
       " 'KAD1',\n",
       " 'KAD2',\n",
       " 'KALRN',\n",
       " 'KAPCA',\n",
       " 'KAPCB',\n",
       " 'KAPCG',\n",
       " 'KC1A',\n",
       " 'KC1AL',\n",
       " 'KC1D',\n",
       " 'KC1E',\n",
       " 'KC1G1',\n",
       " 'KC1G2',\n",
       " 'KC1G3',\n",
       " 'KCC1A',\n",
       " 'KCC1B',\n",
       " 'KCC1D',\n",
       " 'KCC1G',\n",
       " 'KCC2A',\n",
       " 'KCC2B',\n",
       " 'KCC2D',\n",
       " 'KCC2G',\n",
       " 'KCC4',\n",
       " 'KCRM',\n",
       " 'KCRS',\n",
       " 'KGP1',\n",
       " 'KGP2',\n",
       " 'KHK',\n",
       " 'KIT',\n",
       " 'KITM',\n",
       " 'KKCC1',\n",
       " 'KKCC2',\n",
       " 'KPCA',\n",
       " 'KPCB',\n",
       " 'KPCD',\n",
       " 'KPCD1',\n",
       " 'KPCD2',\n",
       " 'KPCD3',\n",
       " 'KPCE',\n",
       " 'KPCG',\n",
       " 'KPCI',\n",
       " 'KPCL',\n",
       " 'KPCT',\n",
       " 'KPCZ',\n",
       " 'KPSH1',\n",
       " 'KPSH2',\n",
       " 'KPYM',\n",
       " 'KS6A1',\n",
       " 'KS6A2',\n",
       " 'KS6A3',\n",
       " 'KS6A4',\n",
       " 'KS6A5',\n",
       " 'KS6A6',\n",
       " 'KS6B1',\n",
       " 'KS6B2',\n",
       " 'KS6C1',\n",
       " 'KSR1',\n",
       " 'KSR2',\n",
       " 'KSYK',\n",
       " 'KT3K',\n",
       " 'LATS1',\n",
       " 'LATS2',\n",
       " 'LCK',\n",
       " 'LIMK1',\n",
       " 'LIMK2',\n",
       " 'LMTK1',\n",
       " 'LMTK2',\n",
       " 'LMTK3',\n",
       " 'LRRK1',\n",
       " 'LRRK2',\n",
       " 'LTK',\n",
       " 'LYN',\n",
       " 'M3K1',\n",
       " 'M3K10',\n",
       " 'M3K11',\n",
       " 'M3K12',\n",
       " 'M3K13',\n",
       " 'M3K14',\n",
       " 'M3K15',\n",
       " 'M3K19',\n",
       " 'M3K2',\n",
       " 'M3K3',\n",
       " 'M3K4',\n",
       " 'M3K5',\n",
       " 'M3K6',\n",
       " 'M3K7',\n",
       " 'M3K8',\n",
       " 'M3K9',\n",
       " 'M3KL4',\n",
       " 'M4K1',\n",
       " 'M4K2',\n",
       " 'M4K3',\n",
       " 'M4K4',\n",
       " 'M4K5',\n",
       " 'MAK',\n",
       " 'MAPK2',\n",
       " 'MAPK3',\n",
       " 'MAPK5',\n",
       " 'MARK1',\n",
       " 'MARK2',\n",
       " 'MARK3',\n",
       " 'MARK4',\n",
       " 'MAST1',\n",
       " 'MAST2',\n",
       " 'MAST3',\n",
       " 'MAST4',\n",
       " 'MATK',\n",
       " 'MELK',\n",
       " 'MERTK',\n",
       " 'MET',\n",
       " 'MINK1',\n",
       " 'MK01',\n",
       " 'MK03',\n",
       " 'MK04',\n",
       " 'MK06',\n",
       " 'MK07',\n",
       " 'MK08',\n",
       " 'MK09',\n",
       " 'MK10',\n",
       " 'MK11',\n",
       " 'MK12',\n",
       " 'MK13',\n",
       " 'MK14',\n",
       " 'MK15',\n",
       " 'MKNK1',\n",
       " 'MKNK2',\n",
       " 'MLKL',\n",
       " 'MLTK',\n",
       " 'MOK',\n",
       " 'MOS',\n",
       " 'MP2K1',\n",
       " 'MP2K2',\n",
       " 'MP2K3',\n",
       " 'MP2K4',\n",
       " 'MP2K5',\n",
       " 'MP2K6',\n",
       " 'MP2K7',\n",
       " 'MRCKA',\n",
       " 'MRCKB',\n",
       " 'MRCKG',\n",
       " 'MST4',\n",
       " 'MTOR',\n",
       " 'MUSK',\n",
       " 'MYLK',\n",
       " 'MYLK2',\n",
       " 'MYLK3',\n",
       " 'MYLK4',\n",
       " 'MYO3A',\n",
       " 'MYO3B',\n",
       " 'NDK3',\n",
       " 'NDK5',\n",
       " 'NDK6',\n",
       " 'NDK7',\n",
       " 'NDKA',\n",
       " 'NDKB',\n",
       " 'NDKM',\n",
       " 'NEK1',\n",
       " 'NEK10',\n",
       " 'NEK11',\n",
       " 'NEK2',\n",
       " 'NEK3',\n",
       " 'NEK4',\n",
       " 'NEK5',\n",
       " 'NEK6',\n",
       " 'NEK7',\n",
       " 'NEK8',\n",
       " 'NEK9',\n",
       " 'NIM1',\n",
       " 'NLK',\n",
       " 'NRBP',\n",
       " 'NRBP2',\n",
       " 'NRK',\n",
       " 'NTKL',\n",
       " 'NTRK1',\n",
       " 'NTRK2',\n",
       " 'NTRK3',\n",
       " 'NUAK1',\n",
       " 'NUAK2',\n",
       " 'OBSCN',\n",
       " 'OXSR1',\n",
       " 'P3C2A',\n",
       " 'P4K2B',\n",
       " 'PACE1',\n",
       " 'PAK1',\n",
       " 'PAK2',\n",
       " 'PAK3',\n",
       " 'PAK4',\n",
       " 'PAK6',\n",
       " 'PAK7',\n",
       " 'PAN3',\n",
       " 'PASK',\n",
       " 'PCKGM',\n",
       " 'PDK1',\n",
       " 'PDK1L',\n",
       " 'PDK2',\n",
       " 'PDK3',\n",
       " 'PDK4',\n",
       " 'PDPK1',\n",
       " 'PDPK2',\n",
       " 'PDXK',\n",
       " 'PEAK1',\n",
       " 'PGFRA',\n",
       " 'PGFRB',\n",
       " 'PGK1',\n",
       " 'PGK2',\n",
       " 'PHKG1',\n",
       " 'PHKG2',\n",
       " 'PI3R4',\n",
       " 'PI42A',\n",
       " 'PI42B',\n",
       " 'PI42C',\n",
       " 'PI4KA',\n",
       " 'PI51A',\n",
       " 'PI51B',\n",
       " 'PI51C',\n",
       " 'PIM1',\n",
       " 'PIM2',\n",
       " 'PIM3',\n",
       " 'PINK1',\n",
       " 'PK3C3',\n",
       " 'PK3CA',\n",
       " 'PK3CB',\n",
       " 'PK3CD',\n",
       " 'PK3CG',\n",
       " 'PKDCC',\n",
       " 'PKN1',\n",
       " 'PKN2',\n",
       " 'PKN3',\n",
       " 'PLK1',\n",
       " 'PLK2',\n",
       " 'PLK3',\n",
       " 'PLK4',\n",
       " 'PLK5',\n",
       " 'PMYT1',\n",
       " 'PRKDC',\n",
       " 'PRKX',\n",
       " 'PRKY',\n",
       " 'PRP4B',\n",
       " 'PRPK',\n",
       " 'PTK6',\n",
       " 'PTK7',\n",
       " 'PXK',\n",
       " 'RAF1',\n",
       " 'RET',\n",
       " 'RIOK1',\n",
       " 'RIOK2',\n",
       " 'RIOK3',\n",
       " 'RIPK1',\n",
       " 'RIPK2',\n",
       " 'RIPK3',\n",
       " 'RIPK4',\n",
       " 'RK',\n",
       " 'RN5A',\n",
       " 'ROCK1',\n",
       " 'ROCK2',\n",
       " 'RON',\n",
       " 'ROR1',\n",
       " 'ROR2',\n",
       " 'ROS1',\n",
       " 'RPKL1',\n",
       " 'RYK',\n",
       " 'SBK1',\n",
       " 'SBK2',\n",
       " 'SCYL2',\n",
       " 'SG110',\n",
       " 'SG196',\n",
       " 'SG223',\n",
       " 'SG494',\n",
       " 'SGK1',\n",
       " 'SGK2',\n",
       " 'SGK3',\n",
       " 'SGK71',\n",
       " 'SIK1',\n",
       " 'SIK2',\n",
       " 'SIK3',\n",
       " 'SLK',\n",
       " 'SMG1',\n",
       " 'SNRK',\n",
       " 'SPEG',\n",
       " 'SRC',\n",
       " 'SRMS',\n",
       " 'SRPK1',\n",
       " 'SRPK2',\n",
       " 'SRPK3',\n",
       " 'ST17A',\n",
       " 'ST17B',\n",
       " 'ST32A',\n",
       " 'ST32B',\n",
       " 'ST32C',\n",
       " 'ST38L',\n",
       " 'STK10',\n",
       " 'STK11',\n",
       " 'STK16',\n",
       " 'STK19',\n",
       " 'STK24',\n",
       " 'STK25',\n",
       " 'STK3',\n",
       " 'STK31',\n",
       " 'STK33',\n",
       " 'STK35',\n",
       " 'STK36',\n",
       " 'STK38',\n",
       " 'STK39',\n",
       " 'STK4',\n",
       " 'STK40',\n",
       " 'STK6',\n",
       " 'STRAA',\n",
       " 'STRAB',\n",
       " 'STYK1',\n",
       " 'TAOK1',\n",
       " 'TAOK2',\n",
       " 'TAOK3',\n",
       " 'TBCK',\n",
       " 'TBK1',\n",
       " 'TEC',\n",
       " 'TESK1',\n",
       " 'TESK2',\n",
       " 'TEX14',\n",
       " 'TGFR1',\n",
       " 'TGFR2',\n",
       " 'TIE1',\n",
       " 'TIE2',\n",
       " 'TIF1A',\n",
       " 'TIF1B',\n",
       " 'TITIN',\n",
       " 'TLK1',\n",
       " 'TLK2',\n",
       " 'TNI3K',\n",
       " 'TNIK',\n",
       " 'TNK1',\n",
       " 'TOPK',\n",
       " 'TRIB1',\n",
       " 'TRIB2',\n",
       " 'TRIB3',\n",
       " 'TRIO',\n",
       " 'TRPM6',\n",
       " 'TRPM7',\n",
       " 'TSSK1',\n",
       " 'TSSK2',\n",
       " 'TSSK3',\n",
       " 'TSSK4',\n",
       " 'TSSK6',\n",
       " 'TTBK1',\n",
       " 'TTBK2',\n",
       " 'TTK',\n",
       " 'TXK',\n",
       " 'TYK2',\n",
       " 'TYRO3',\n",
       " 'UFO',\n",
       " 'UHMK1',\n",
       " 'ULK1',\n",
       " 'ULK2',\n",
       " 'ULK3',\n",
       " 'ULK4',\n",
       " 'VGFR1',\n",
       " 'VGFR2',\n",
       " 'VGFR3',\n",
       " 'VIP1',\n",
       " 'VIP2',\n",
       " 'VRK1',\n",
       " 'VRK2',\n",
       " 'VRK3',\n",
       " 'WEE1',\n",
       " 'WEE2',\n",
       " 'WNK1',\n",
       " 'WNK2',\n",
       " 'WNK3',\n",
       " 'WNK4',\n",
       " 'YES',\n",
       " 'ZAP70'}"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uniprot_targets = set(kmast['Uniprot Protein'])\n",
    "uniprot_targets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "146363\n"
     ]
    }
   ],
   "source": [
    "print(len(uni_mapper))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Start importing sources"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_sources = {}\n",
    "data_preproc = {}\n",
    "\n",
    "# hardcode MAPK3 due to MAPKAPK3 issues\n",
    "id_dict = lambda x: hgnc_mapper[x]  if x in hgnc_mapper.keys()\\\n",
    "                                        else 'MAPK3' if x == 'P27361'\\\n",
    "                                        else uniprot_to_hgnc_map_helper[x] if x in uniprot_to_hgnc_map_helper.keys()\\\n",
    "                                        else uniprot_to_hgnc_map_helper[uni_mapper[x]] if x in uni_mapper.keys() and uni_mapper[x] in uniprot_to_hgnc_map_helper.keys()\\\n",
    "                                        else hgnc_mapper[uni_mapper[x]] if x in uni_mapper.keys() and uni_mapper[x] in hgnc_mapper.keys()\\\n",
    "                                        else None\n",
    "\n",
    "#else uniprot_to_hgnc_map_helper[uni_mapper[entrez_to_uniprot_number[x]]] if x in entrez_to_uniprot_number.keys() and entrez_to_uniprot_number[x] in uni_mapper.keys() and uni_mapper[entrez_to_uniprot_number[x]] in uniprot_to_hgnc_map_helper.keys()\\\n",
    "#else uniprot_to_hgnc_map_helper[uni_mapper[ensg_to_uniprot_number[x]]] if x in ensg_to_uniprot_number.keys() and ensg_to_uniprot_number[x] in uni_mapper.keys() and uni_mapper[ensg_to_uniprot_number[x]] in uniprot_to_hgnc_map_helper.keys()\\\n",
    "                                        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Int64Index([0, 1, 2, 3, 4, 5], dtype='int64')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AL1A1_HUMAN</td>\n",
       "      <td>216</td>\n",
       "      <td>AL1A1_HUMAN</td>\n",
       "      <td>216</td>\n",
       "      <td>0.76</td>\n",
       "      <td>experiments:in vivo,Two-hybrid;pmids:12081471,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ITA7_HUMAN</td>\n",
       "      <td>3679</td>\n",
       "      <td>ACHA_HUMAN</td>\n",
       "      <td>1134</td>\n",
       "      <td>0.73</td>\n",
       "      <td>experiments:in vivo,Affinity Capture-Western,a...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NEB1_HUMAN</td>\n",
       "      <td>55607</td>\n",
       "      <td>ACTG_HUMAN</td>\n",
       "      <td>71</td>\n",
       "      <td>0.65</td>\n",
       "      <td>experiments:in vitro,in vivo;pmids:9362513,120...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRGN_HUMAN</td>\n",
       "      <td>5552</td>\n",
       "      <td>CD44_HUMAN</td>\n",
       "      <td>960</td>\n",
       "      <td>0.63</td>\n",
       "      <td>experiments:in vivo;pmids:9334256,16189514,167...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GRB7_HUMAN</td>\n",
       "      <td>2886</td>\n",
       "      <td>ERBB2_HUMAN</td>\n",
       "      <td>2064</td>\n",
       "      <td>0.90</td>\n",
       "      <td>experiments:in vitro,in vivo,Reconstituted Com...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             0      1            2     3     4  \\\n",
       "0  AL1A1_HUMAN    216  AL1A1_HUMAN   216  0.76   \n",
       "1   ITA7_HUMAN   3679   ACHA_HUMAN  1134  0.73   \n",
       "2   NEB1_HUMAN  55607   ACTG_HUMAN    71  0.65   \n",
       "3   SRGN_HUMAN   5552   CD44_HUMAN   960  0.63   \n",
       "4   GRB7_HUMAN   2886  ERBB2_HUMAN  2064  0.90   \n",
       "\n",
       "                                                   5  \n",
       "0  experiments:in vivo,Two-hybrid;pmids:12081471,...  \n",
       "1  experiments:in vivo,Affinity Capture-Western,a...  \n",
       "2  experiments:in vitro,in vivo;pmids:9362513,120...  \n",
       "3  experiments:in vivo;pmids:9334256,16189514,167...  \n",
       "4  experiments:in vitro,in vivo,Reconstituted Com...  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hippie = pd.read_csv('../data/network/hippie_current.txt', sep='\\t', header=None)\n",
    "\n",
    "data_sources['hippie'] = hippie\n",
    "data_preproc['hippie'] = {\n",
    "    'columns':[0,2],\n",
    "    'maps':[\n",
    "        lambda x: id_dict(x.split('_')[0]),\n",
    "        lambda x: id_dict(x.split('_')[0])\n",
    "    ]\n",
    "}\n",
    "\n",
    "print(hippie.columns)\n",
    "hippie.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['GENE', 'KINASE', 'KIN_ACC_ID', 'KIN_ORGANISM', 'SUBSTRATE',\n",
      "       'SUB_GENE_ID', 'SUB_ACC_ID', 'SUB_GENE', 'SUB_ORGANISM', 'SUB_MOD_RSD',\n",
      "       'SITE_GRP_ID', 'SITE_+/-7_AA', 'DOMAIN', 'IN_VIVO_RXN', 'IN_VITRO_RXN',\n",
      "       'CST_CAT#'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>KINASE</th>\n",
       "      <th>KIN_ACC_ID</th>\n",
       "      <th>KIN_ORGANISM</th>\n",
       "      <th>SUBSTRATE</th>\n",
       "      <th>SUB_GENE_ID</th>\n",
       "      <th>SUB_ACC_ID</th>\n",
       "      <th>SUB_GENE</th>\n",
       "      <th>SUB_ORGANISM</th>\n",
       "      <th>SUB_MOD_RSD</th>\n",
       "      <th>SITE_GRP_ID</th>\n",
       "      <th>SITE_+/-7_AA</th>\n",
       "      <th>DOMAIN</th>\n",
       "      <th>IN_VIVO_RXN</th>\n",
       "      <th>IN_VITRO_RXN</th>\n",
       "      <th>CST_CAT#</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Pak2</td>\n",
       "      <td>PAK2</td>\n",
       "      <td>Q64303</td>\n",
       "      <td>rat</td>\n",
       "      <td>MEK1</td>\n",
       "      <td>170851.0</td>\n",
       "      <td>Q01986</td>\n",
       "      <td>Map2k1</td>\n",
       "      <td>rat</td>\n",
       "      <td>S298</td>\n",
       "      <td>448284</td>\n",
       "      <td>RtPGRPLsSYGMDSR</td>\n",
       "      <td>Pkinase</td>\n",
       "      <td></td>\n",
       "      <td>X</td>\n",
       "      <td>9128; 98195</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Pak2</td>\n",
       "      <td>PAK2</td>\n",
       "      <td>Q64303</td>\n",
       "      <td>rat</td>\n",
       "      <td>PRKD1</td>\n",
       "      <td>85421.0</td>\n",
       "      <td>Q9WTQ1</td>\n",
       "      <td>Prkd1</td>\n",
       "      <td>rat</td>\n",
       "      <td>S203</td>\n",
       "      <td>449896</td>\n",
       "      <td>GVRRRRLsNVsLTGL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Pak2</td>\n",
       "      <td>PAK2</td>\n",
       "      <td>Q64303</td>\n",
       "      <td>rat</td>\n",
       "      <td>prolactin</td>\n",
       "      <td>5617.0</td>\n",
       "      <td>P01236</td>\n",
       "      <td>PRL</td>\n",
       "      <td>human</td>\n",
       "      <td>S207</td>\n",
       "      <td>451732</td>\n",
       "      <td>LHCLRRDsHKIDNYL</td>\n",
       "      <td>Hormone_1</td>\n",
       "      <td></td>\n",
       "      <td>X</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Pak2</td>\n",
       "      <td>PAK2</td>\n",
       "      <td>Q64303</td>\n",
       "      <td>rat</td>\n",
       "      <td>prolactin</td>\n",
       "      <td>24683.0</td>\n",
       "      <td>P01237</td>\n",
       "      <td>Prl</td>\n",
       "      <td>rat</td>\n",
       "      <td>S206</td>\n",
       "      <td>451732</td>\n",
       "      <td>IRCLRRDsHKVDNYL</td>\n",
       "      <td>Hormone_1</td>\n",
       "      <td></td>\n",
       "      <td>X</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>EIF2AK1</td>\n",
       "      <td>HRI</td>\n",
       "      <td>Q9BQI3</td>\n",
       "      <td>human</td>\n",
       "      <td>eIF2-alpha</td>\n",
       "      <td>54318.0</td>\n",
       "      <td>P68101</td>\n",
       "      <td>Eif2s1</td>\n",
       "      <td>rat</td>\n",
       "      <td>S52</td>\n",
       "      <td>447635</td>\n",
       "      <td>MILLSELsRRRIRSI</td>\n",
       "      <td>S1</td>\n",
       "      <td></td>\n",
       "      <td>X</td>\n",
       "      <td>3597; 9721; 3398; 5199</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      GENE KINASE KIN_ACC_ID KIN_ORGANISM   SUBSTRATE  SUB_GENE_ID SUB_ACC_ID  \\\n",
       "0     Pak2   PAK2     Q64303          rat        MEK1     170851.0     Q01986   \n",
       "1     Pak2   PAK2     Q64303          rat       PRKD1      85421.0     Q9WTQ1   \n",
       "2     Pak2   PAK2     Q64303          rat   prolactin       5617.0     P01236   \n",
       "3     Pak2   PAK2     Q64303          rat   prolactin      24683.0     P01237   \n",
       "4  EIF2AK1    HRI     Q9BQI3        human  eIF2-alpha      54318.0     P68101   \n",
       "\n",
       "  SUB_GENE SUB_ORGANISM SUB_MOD_RSD  SITE_GRP_ID     SITE_+/-7_AA     DOMAIN  \\\n",
       "0   Map2k1          rat        S298       448284  RtPGRPLsSYGMDSR    Pkinase   \n",
       "1    Prkd1          rat        S203       449896  GVRRRRLsNVsLTGL        NaN   \n",
       "2      PRL        human        S207       451732  LHCLRRDsHKIDNYL  Hormone_1   \n",
       "3      Prl          rat        S206       451732  IRCLRRDsHKVDNYL  Hormone_1   \n",
       "4   Eif2s1          rat         S52       447635  MILLSELsRRRIRSI         S1   \n",
       "\n",
       "  IN_VIVO_RXN IN_VITRO_RXN                CST_CAT#  \n",
       "0                        X             9128; 98195  \n",
       "1           X                                  NaN  \n",
       "2                        X                     NaN  \n",
       "3                        X                     NaN  \n",
       "4                        X  3597; 9721; 3398; 5199  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "phosphosite_substrate = pd.read_csv('../data/network/Kinase_Substrate_Dataset.txt', sep='\\t', skiprows=2)\n",
    "\n",
    "data_sources['phosphosite_substrate'] = phosphosite_substrate\n",
    "data_preproc['phosphosite_substrate'] = {\n",
    "    'columns':['GENE','SUB_GENE'],\n",
    "    'maps':[\n",
    "        lambda x: id_dict(x.upper()),\n",
    "        lambda x: id_dict(x.upper())\n",
    "    ]\n",
    "}\n",
    "\n",
    "print(phosphosite_substrate.columns)\n",
    "display(phosphosite_substrate.head())\n",
    "\n",
    "\n",
    "phosphosite_regulatory = pd.read_csv('../data/network/Regulatory_sites.txt', sep='\\t', skiprows=3, error_bad_lines=False)\n",
    "\n",
    "#data_sources['phosphosite_regulatory'] =  phosphosite_regulatory\n",
    "data_preproc['phosphosite_regulatory'] = {\n",
    "    'columns':['GENE','SUBSTRATE'],\n",
    "    'maps':[\n",
    "        lambda x: id_dict(x.upper()),\n",
    "        lambda x: id_dict(x.upper())\n",
    "    ]\n",
    "}\n",
    "\n",
    "#print(phosphosite_regulatory.columns)\n",
    "#display(phosphosite_regulatory.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Int64Index([0, 1, 2, 3, 4, 5, 6, 7], dtype='int64')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ALDH1A1</td>\n",
       "      <td>1</td>\n",
       "      <td>NP_000680.2</td>\n",
       "      <td>ALDH1A1</td>\n",
       "      <td>1</td>\n",
       "      <td>NP_000680.2</td>\n",
       "      <td>in vivo;yeast 2-hybrid</td>\n",
       "      <td>1,208,147,116,189,510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ITGA7</td>\n",
       "      <td>2761</td>\n",
       "      <td>NP_001138468.1</td>\n",
       "      <td>CHRNA1</td>\n",
       "      <td>7</td>\n",
       "      <td>NP_001034612.1</td>\n",
       "      <td>in vivo</td>\n",
       "      <td>10910772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PPP1R9A</td>\n",
       "      <td>16000</td>\n",
       "      <td>NP_060120.2</td>\n",
       "      <td>ACTG1</td>\n",
       "      <td>17</td>\n",
       "      <td>NP_001605.1</td>\n",
       "      <td>in vitro;in vivo</td>\n",
       "      <td>936,251,312,052,877</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRGN</td>\n",
       "      <td>1513</td>\n",
       "      <td>NP_002718.2</td>\n",
       "      <td>CD44</td>\n",
       "      <td>115</td>\n",
       "      <td>NP_000601.3</td>\n",
       "      <td>in vivo</td>\n",
       "      <td>9334256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GRB7</td>\n",
       "      <td>3311</td>\n",
       "      <td>NP_005301.2</td>\n",
       "      <td>ERBB2</td>\n",
       "      <td>1281</td>\n",
       "      <td>NP_004439.2</td>\n",
       "      <td>in vitro;in vivo</td>\n",
       "      <td>9079677</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         0      1               2        3     4               5  \\\n",
       "0  ALDH1A1      1     NP_000680.2  ALDH1A1     1     NP_000680.2   \n",
       "1    ITGA7   2761  NP_001138468.1   CHRNA1     7  NP_001034612.1   \n",
       "2  PPP1R9A  16000     NP_060120.2    ACTG1    17     NP_001605.1   \n",
       "3     SRGN   1513     NP_002718.2     CD44   115     NP_000601.3   \n",
       "4     GRB7   3311     NP_005301.2    ERBB2  1281     NP_004439.2   \n",
       "\n",
       "                        6                      7  \n",
       "0  in vivo;yeast 2-hybrid  1,208,147,116,189,510  \n",
       "1                 in vivo               10910772  \n",
       "2        in vitro;in vivo    936,251,312,052,877  \n",
       "3                 in vivo                9334256  \n",
       "4        in vitro;in vivo                9079677  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hprd = pd.read_csv('../data/network/HPRD_ALL_BINARY_PROTEIN_PROTEIN_INTERACTIONS.txt', sep='\\t', header=None)\n",
    "\n",
    "data_sources['hprd'] = hprd\n",
    "data_preproc['hprd'] = {\n",
    "    'columns':[0,3],\n",
    "    'maps':[\n",
    "        lambda x: id_dict(x.upper()),\n",
    "        lambda x: id_dict(x.upper())\n",
    "    ]\n",
    "}\n",
    "\n",
    "print(hprd.columns)\n",
    "hprd.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Q7Z5W8'"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "entrez_to_uniprot_number['31']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['# Interactor 1 uniprot id', 'Interactor 1 Ensembl gene id',\n",
      "       'Interactor 1 Entrez Gene id', 'Interactor 2 uniprot id',\n",
      "       'Interactor 2 Ensembl gene id', 'Interactor 2 Entrez Gene id',\n",
      "       'Interaction type', 'Interaction context', 'Pubmed references'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th># Interactor 1 uniprot id</th>\n",
       "      <th>Interactor 1 Ensembl gene id</th>\n",
       "      <th>Interactor 1 Entrez Gene id</th>\n",
       "      <th>Interactor 2 uniprot id</th>\n",
       "      <th>Interactor 2 Ensembl gene id</th>\n",
       "      <th>Interactor 2 Entrez Gene id</th>\n",
       "      <th>Interaction type</th>\n",
       "      <th>Interaction context</th>\n",
       "      <th>Pubmed references</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>uniprotkb:Q9Y287</td>\n",
       "      <td>ENSEMBL:ENSG00000136156</td>\n",
       "      <td>entrezgene/locuslink:9445</td>\n",
       "      <td>uniprotkb:Q9Y287</td>\n",
       "      <td>ENSEMBL:ENSG00000136156</td>\n",
       "      <td>entrezgene/locuslink:9445</td>\n",
       "      <td>physical association</td>\n",
       "      <td>reactome:R-HSA-976871</td>\n",
       "      <td>14690516|10391242</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>uniprotkb:P37840</td>\n",
       "      <td>ENSEMBL:ENSG00000145335</td>\n",
       "      <td>entrezgene/locuslink:6622</td>\n",
       "      <td>uniprotkb:P37840</td>\n",
       "      <td>ENSEMBL:ENSG00000145335</td>\n",
       "      <td>entrezgene/locuslink:6622</td>\n",
       "      <td>physical association</td>\n",
       "      <td>reactome:R-HSA-1247852</td>\n",
       "      <td>24243840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>uniprotkb:P0DJI8</td>\n",
       "      <td>ENSEMBL:ENSG00000173432</td>\n",
       "      <td>entrezgene/locuslink:6288</td>\n",
       "      <td>uniprotkb:P0DJI8</td>\n",
       "      <td>ENSEMBL:ENSG00000173432</td>\n",
       "      <td>entrezgene/locuslink:6288</td>\n",
       "      <td>physical association</td>\n",
       "      <td>reactome:R-HSA-976898</td>\n",
       "      <td>19393650|103558</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>uniprotkb:P06727</td>\n",
       "      <td>ENSEMBL:ENSG00000110244</td>\n",
       "      <td>entrezgene/locuslink:337</td>\n",
       "      <td>uniprotkb:P06727</td>\n",
       "      <td>ENSEMBL:ENSG00000110244</td>\n",
       "      <td>entrezgene/locuslink:337</td>\n",
       "      <td>physical association</td>\n",
       "      <td>reactome:R-HSA-976889</td>\n",
       "      <td>15146166</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>uniprotkb:P01160</td>\n",
       "      <td>ENSEMBL:ENSG00000175206</td>\n",
       "      <td>entrezgene/locuslink:4878</td>\n",
       "      <td>uniprotkb:P01160</td>\n",
       "      <td>ENSEMBL:ENSG00000175206</td>\n",
       "      <td>entrezgene/locuslink:4878</td>\n",
       "      <td>physical association</td>\n",
       "      <td>reactome:R-HSA-976987</td>\n",
       "      <td>2142465|2945573</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  # Interactor 1 uniprot id Interactor 1 Ensembl gene id  \\\n",
       "0          uniprotkb:Q9Y287      ENSEMBL:ENSG00000136156   \n",
       "1          uniprotkb:P37840      ENSEMBL:ENSG00000145335   \n",
       "2          uniprotkb:P0DJI8      ENSEMBL:ENSG00000173432   \n",
       "3          uniprotkb:P06727      ENSEMBL:ENSG00000110244   \n",
       "4          uniprotkb:P01160      ENSEMBL:ENSG00000175206   \n",
       "\n",
       "  Interactor 1 Entrez Gene id Interactor 2 uniprot id  \\\n",
       "0   entrezgene/locuslink:9445        uniprotkb:Q9Y287   \n",
       "1   entrezgene/locuslink:6622        uniprotkb:P37840   \n",
       "2   entrezgene/locuslink:6288        uniprotkb:P0DJI8   \n",
       "3    entrezgene/locuslink:337        uniprotkb:P06727   \n",
       "4   entrezgene/locuslink:4878        uniprotkb:P01160   \n",
       "\n",
       "  Interactor 2 Ensembl gene id Interactor 2 Entrez Gene id  \\\n",
       "0      ENSEMBL:ENSG00000136156   entrezgene/locuslink:9445   \n",
       "1      ENSEMBL:ENSG00000145335   entrezgene/locuslink:6622   \n",
       "2      ENSEMBL:ENSG00000173432   entrezgene/locuslink:6288   \n",
       "3      ENSEMBL:ENSG00000110244    entrezgene/locuslink:337   \n",
       "4      ENSEMBL:ENSG00000175206   entrezgene/locuslink:4878   \n",
       "\n",
       "       Interaction type     Interaction context  Pubmed references  \n",
       "0  physical association   reactome:R-HSA-976871  14690516|10391242  \n",
       "1  physical association  reactome:R-HSA-1247852           24243840  \n",
       "2  physical association   reactome:R-HSA-976898    19393650|103558  \n",
       "3  physical association   reactome:R-HSA-976889           15146166  \n",
       "4  physical association   reactome:R-HSA-976987    2142465|2945573  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reactome = pd.read_csv('../data/network/reactome.homo_sapiens.interactions.tab-delimited.txt', sep='\\t')\n",
    "\n",
    "data_sources['reactome'] = reactome\n",
    "data_preproc['reactome'] = {\n",
    "    'columns': ['# Interactor 1 uniprot id','Interactor 2 uniprot id'],\n",
    "    'maps': [\n",
    "        lambda x: id_dict(x.split(':')[1]),\n",
    "        lambda x: id_dict(x.split(':')[1])\n",
    "    ]\n",
    "}\n",
    "\n",
    "\n",
    "print(reactome.columns)\n",
    "reactome.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['#BioGRID Interaction ID', 'Entrez Gene Interactor A',\n",
      "       'Entrez Gene Interactor B', 'BioGRID ID Interactor A',\n",
      "       'BioGRID ID Interactor B', 'Systematic Name Interactor A',\n",
      "       'Systematic Name Interactor B', 'Official Symbol Interactor A',\n",
      "       'Official Symbol Interactor B', 'Synonyms Interactor A',\n",
      "       'Synonyms Interactor B', 'Experimental System',\n",
      "       'Experimental System Type', 'Author', 'Pubmed ID',\n",
      "       'Organism Interactor A', 'Organism Interactor B', 'Throughput', 'Score',\n",
      "       'Modification', 'Phenotypes', 'Qualifications', 'Tags',\n",
      "       'Source Database'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#BioGRID Interaction ID</th>\n",
       "      <th>Entrez Gene Interactor A</th>\n",
       "      <th>Entrez Gene Interactor B</th>\n",
       "      <th>BioGRID ID Interactor A</th>\n",
       "      <th>BioGRID ID Interactor B</th>\n",
       "      <th>Systematic Name Interactor A</th>\n",
       "      <th>Systematic Name Interactor B</th>\n",
       "      <th>Official Symbol Interactor A</th>\n",
       "      <th>Official Symbol Interactor B</th>\n",
       "      <th>Synonyms Interactor A</th>\n",
       "      <th>...</th>\n",
       "      <th>Pubmed ID</th>\n",
       "      <th>Organism Interactor A</th>\n",
       "      <th>Organism Interactor B</th>\n",
       "      <th>Throughput</th>\n",
       "      <th>Score</th>\n",
       "      <th>Modification</th>\n",
       "      <th>Phenotypes</th>\n",
       "      <th>Qualifications</th>\n",
       "      <th>Tags</th>\n",
       "      <th>Source Database</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>103</td>\n",
       "      <td>6416</td>\n",
       "      <td>2318</td>\n",
       "      <td>112315</td>\n",
       "      <td>108607</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>MAP2K4</td>\n",
       "      <td>FLNC</td>\n",
       "      <td>JNKK|JNKK1|MAPKK4|MEK4|MKK4|PRKMK4|SAPKK-1|SAP...</td>\n",
       "      <td>...</td>\n",
       "      <td>9006895</td>\n",
       "      <td>9606</td>\n",
       "      <td>9606</td>\n",
       "      <td>Low Throughput</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>BIOGRID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>117</td>\n",
       "      <td>84665</td>\n",
       "      <td>88</td>\n",
       "      <td>124185</td>\n",
       "      <td>106603</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>MYPN</td>\n",
       "      <td>ACTN2</td>\n",
       "      <td>CMD1DD|CMH22|MYOP|RCM4</td>\n",
       "      <td>...</td>\n",
       "      <td>11309420</td>\n",
       "      <td>9606</td>\n",
       "      <td>9606</td>\n",
       "      <td>Low Throughput</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>BIOGRID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>183</td>\n",
       "      <td>90</td>\n",
       "      <td>2339</td>\n",
       "      <td>106605</td>\n",
       "      <td>108625</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ACVR1</td>\n",
       "      <td>FNTA</td>\n",
       "      <td>ACTRI|ACVR1A|ACVRLK2|ALK2|FOP|SKR1|TSRI</td>\n",
       "      <td>...</td>\n",
       "      <td>8599089</td>\n",
       "      <td>9606</td>\n",
       "      <td>9606</td>\n",
       "      <td>Low Throughput</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>BIOGRID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>278</td>\n",
       "      <td>2624</td>\n",
       "      <td>5371</td>\n",
       "      <td>108894</td>\n",
       "      <td>111384</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>GATA2</td>\n",
       "      <td>PML</td>\n",
       "      <td>DCML|IMD21|MONOMAC|NFE1B</td>\n",
       "      <td>...</td>\n",
       "      <td>10938104</td>\n",
       "      <td>9606</td>\n",
       "      <td>9606</td>\n",
       "      <td>Low Throughput</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>BIOGRID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>418</td>\n",
       "      <td>6118</td>\n",
       "      <td>6774</td>\n",
       "      <td>112038</td>\n",
       "      <td>112651</td>\n",
       "      <td>RP4-547C9.3</td>\n",
       "      <td>-</td>\n",
       "      <td>RPA2</td>\n",
       "      <td>STAT3</td>\n",
       "      <td>REPA2|RP-A p32|RP-A p34|RPA32</td>\n",
       "      <td>...</td>\n",
       "      <td>10875894</td>\n",
       "      <td>9606</td>\n",
       "      <td>9606</td>\n",
       "      <td>Low Throughput</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>BIOGRID</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   #BioGRID Interaction ID  Entrez Gene Interactor A  \\\n",
       "0                      103                      6416   \n",
       "1                      117                     84665   \n",
       "2                      183                        90   \n",
       "3                      278                      2624   \n",
       "4                      418                      6118   \n",
       "\n",
       "   Entrez Gene Interactor B  BioGRID ID Interactor A  BioGRID ID Interactor B  \\\n",
       "0                      2318                   112315                   108607   \n",
       "1                        88                   124185                   106603   \n",
       "2                      2339                   106605                   108625   \n",
       "3                      5371                   108894                   111384   \n",
       "4                      6774                   112038                   112651   \n",
       "\n",
       "  Systematic Name Interactor A Systematic Name Interactor B  \\\n",
       "0                            -                            -   \n",
       "1                            -                            -   \n",
       "2                            -                            -   \n",
       "3                            -                            -   \n",
       "4                  RP4-547C9.3                            -   \n",
       "\n",
       "  Official Symbol Interactor A Official Symbol Interactor B  \\\n",
       "0                       MAP2K4                         FLNC   \n",
       "1                         MYPN                        ACTN2   \n",
       "2                        ACVR1                         FNTA   \n",
       "3                        GATA2                          PML   \n",
       "4                         RPA2                        STAT3   \n",
       "\n",
       "                               Synonyms Interactor A  ... Pubmed ID  \\\n",
       "0  JNKK|JNKK1|MAPKK4|MEK4|MKK4|PRKMK4|SAPKK-1|SAP...  ...   9006895   \n",
       "1                             CMD1DD|CMH22|MYOP|RCM4  ...  11309420   \n",
       "2            ACTRI|ACVR1A|ACVRLK2|ALK2|FOP|SKR1|TSRI  ...   8599089   \n",
       "3                           DCML|IMD21|MONOMAC|NFE1B  ...  10938104   \n",
       "4                      REPA2|RP-A p32|RP-A p34|RPA32  ...  10875894   \n",
       "\n",
       "  Organism Interactor A Organism Interactor B      Throughput  Score  \\\n",
       "0                  9606                  9606  Low Throughput      -   \n",
       "1                  9606                  9606  Low Throughput      -   \n",
       "2                  9606                  9606  Low Throughput      -   \n",
       "3                  9606                  9606  Low Throughput      -   \n",
       "4                  9606                  9606  Low Throughput      -   \n",
       "\n",
       "   Modification  Phenotypes Qualifications Tags Source Database  \n",
       "0             -           -              -    -         BIOGRID  \n",
       "1             -           -              -    -         BIOGRID  \n",
       "2             -           -              -    -         BIOGRID  \n",
       "3             -           -              -    -         BIOGRID  \n",
       "4             -           -              -    -         BIOGRID  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biogrid = pd.read_csv('../data/network/BIOGRID-ORGANISM-Homo_sapiens-3.5.168.tab2.txt', low_memory=False, sep='\\t')\n",
    "\n",
    "data_sources['biogrid'] = biogrid\n",
    "data_preproc['biogrid'] = {\n",
    "    'columns': ['Official Symbol Interactor A','Official Symbol Interactor B'],\n",
    "    'maps': [\n",
    "        lambda x: id_dict(x),\n",
    "        lambda x: id_dict(x)\n",
    "    ]\n",
    "}\n",
    "\n",
    "print(biogrid.columns)\n",
    "biogrid.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Dataset', 'SwissProt1', 'SwissProt2'], dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Dataset</th>\n",
       "      <th>SwissProt1</th>\n",
       "      <th>SwissProt2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SOURAV_MAPK_LOW</td>\n",
       "      <td>P63000</td>\n",
       "      <td>A0AUZ9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IntAct</td>\n",
       "      <td>Q96CV9</td>\n",
       "      <td>A0AUZ9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BioGrid</td>\n",
       "      <td>P0CG48</td>\n",
       "      <td>A0AV96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IntAct_Mouse</td>\n",
       "      <td>P62258</td>\n",
       "      <td>A0AV96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IntAct_Mouse</td>\n",
       "      <td>P63104</td>\n",
       "      <td>A0AV96</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Dataset SwissProt1 SwissProt2\n",
       "0  SOURAV_MAPK_LOW     P63000     A0AUZ9\n",
       "1           IntAct     Q96CV9     A0AUZ9\n",
       "2          BioGrid     P0CG48     A0AV96\n",
       "3     IntAct_Mouse     P62258     A0AV96\n",
       "4     IntAct_Mouse     P63104     A0AV96"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "i2d = pd.read_csv('../data/network/i2d.2_9.txt', sep='\\t')\n",
    "\n",
    "data_sources['i2d'] = i2d\n",
    "data_preproc['i2d'] = {\n",
    "    'columns': ['SwissProt1','SwissProt2'],\n",
    "    'maps': [\n",
    "        lambda x: id_dict(x),\n",
    "        lambda x: id_dict(x)\n",
    "    ]\n",
    "}\n",
    "\n",
    "print(i2d.columns)\n",
    "i2d.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Int64Index([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14], dtype='int64')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "      <th>12</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>uniprotkb:Q72547</td>\n",
       "      <td>uniprotkb:Q72547</td>\n",
       "      <td>intact:EBI-7484755|intact:MINT-8208544</td>\n",
       "      <td>intact:EBI-7484755|intact:MINT-8208544</td>\n",
       "      <td>psi-mi:q72547_9hiv1(display_long)|uniprotkb:po...</td>\n",
       "      <td>psi-mi:q72547_9hiv1(display_long)|uniprotkb:po...</td>\n",
       "      <td>psi-mi:\"MI:0030\"(cross-linking study)</td>\n",
       "      <td>Nishitsuji et al. (2011)</td>\n",
       "      <td>pubmed:22004763|imex:IM-16791|mint:MINT-820698...</td>\n",
       "      <td>taxid:11676(9hiv1)|taxid:11676(Human immunodef...</td>\n",
       "      <td>taxid:11676(9hiv1)|taxid:11676(Human immunodef...</td>\n",
       "      <td>psi-mi:\"MI:0915\"(physical association)</td>\n",
       "      <td>psi-mi:\"MI:0471\"(MINT)</td>\n",
       "      <td>intact:EBI-7484777|mint:MINT-8206988|imex:IM-1...</td>\n",
       "      <td>intact-miscore:0.40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>uniprotkb:Q72547</td>\n",
       "      <td>uniprotkb:Q72547</td>\n",
       "      <td>intact:EBI-7484755|intact:MINT-8208544</td>\n",
       "      <td>intact:EBI-7484755|intact:MINT-8208544</td>\n",
       "      <td>psi-mi:q72547_9hiv1(display_long)|uniprotkb:po...</td>\n",
       "      <td>psi-mi:q72547_9hiv1(display_long)|uniprotkb:po...</td>\n",
       "      <td>psi-mi:\"MI:0030\"(cross-linking study)</td>\n",
       "      <td>Nishitsuji et al. (2011)</td>\n",
       "      <td>pubmed:22004763|imex:IM-16791|mint:MINT-820698...</td>\n",
       "      <td>taxid:11676(9hiv1)|taxid:11676(Human immunodef...</td>\n",
       "      <td>taxid:11676(9hiv1)|taxid:11676(Human immunodef...</td>\n",
       "      <td>psi-mi:\"MI:0915\"(physical association)</td>\n",
       "      <td>psi-mi:\"MI:0471\"(MINT)</td>\n",
       "      <td>intact:EBI-7484916|mint:MINT-8207003|imex:IM-1...</td>\n",
       "      <td>intact-miscore:0.40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>uniprotkb:Q72547</td>\n",
       "      <td>uniprotkb:Q72547</td>\n",
       "      <td>intact:EBI-7484755|intact:MINT-8208544</td>\n",
       "      <td>intact:EBI-7484755|intact:MINT-8208544</td>\n",
       "      <td>psi-mi:q72547_9hiv1(display_long)|uniprotkb:po...</td>\n",
       "      <td>psi-mi:q72547_9hiv1(display_long)|uniprotkb:po...</td>\n",
       "      <td>psi-mi:\"MI:0030\"(cross-linking study)</td>\n",
       "      <td>Nishitsuji et al. (2011)</td>\n",
       "      <td>pubmed:22004763|imex:IM-16791|mint:MINT-820698...</td>\n",
       "      <td>taxid:11676(9hiv1)|taxid:11676(Human immunodef...</td>\n",
       "      <td>taxid:11676(9hiv1)|taxid:11676(Human immunodef...</td>\n",
       "      <td>psi-mi:\"MI:0915\"(physical association)</td>\n",
       "      <td>psi-mi:\"MI:0471\"(MINT)</td>\n",
       "      <td>intact:EBI-7484965|mint:MINT-8207016|imex:IM-1...</td>\n",
       "      <td>intact-miscore:0.40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>uniprotkb:Q90VU7</td>\n",
       "      <td>uniprotkb:Q900A7</td>\n",
       "      <td>intact:EBI-7460704|intact:MINT-5281653</td>\n",
       "      <td>intact:EBI-7460739|intact:MINT-8208185</td>\n",
       "      <td>psi-mi:q90vu7_9hiv1(display_long)|uniprotkb:ne...</td>\n",
       "      <td>psi-mi:q900a7_9hiv1(display_long)|uniprotkb:ta...</td>\n",
       "      <td>psi-mi:\"MI:0416\"(fluorescence microscopy)</td>\n",
       "      <td>Sugiyama et al. (2011)</td>\n",
       "      <td>pubmed:21970979|imex:IM-16787|mint:MINT-820620...</td>\n",
       "      <td>taxid:11676(9hiv1)|taxid:11676(Human immunodef...</td>\n",
       "      <td>taxid:11676(9hiv1)|taxid:11676(Human immunodef...</td>\n",
       "      <td>psi-mi:\"MI:0403\"(colocalization)</td>\n",
       "      <td>psi-mi:\"MI:0471\"(MINT)</td>\n",
       "      <td>intact:EBI-7460732|mint:MINT-8206271|imex:IM-1...</td>\n",
       "      <td>intact-miscore:0.27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>uniprotkb:Q2Q067</td>\n",
       "      <td>uniprotkb:Q13015</td>\n",
       "      <td>intact:EBI-9675545</td>\n",
       "      <td>intact:EBI-6269719</td>\n",
       "      <td>psi-mi:q2q067_9dela(display_long)|uniprotkb:HB...</td>\n",
       "      <td>psi-mi:af1q_human(display_long)|uniprotkb:MLLT...</td>\n",
       "      <td>psi-mi:\"MI:0397\"(two hybrid array)</td>\n",
       "      <td>Simonis et al. (2012)</td>\n",
       "      <td>imex:IM-22977|pubmed:22458338</td>\n",
       "      <td>taxid:11908(humt-)|taxid:11908(Human T-lymphot...</td>\n",
       "      <td>taxid:9606(human)|taxid:9606(Homo sapiens)</td>\n",
       "      <td>psi-mi:\"MI:0915\"(physical association)</td>\n",
       "      <td>psi-mi:\"MI:0471\"(MINT)</td>\n",
       "      <td>intact:EBI-9675551|imex:IM-22977-4</td>\n",
       "      <td>intact-miscore:0.49</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 0                 1                                       2   \\\n",
       "0  uniprotkb:Q72547  uniprotkb:Q72547  intact:EBI-7484755|intact:MINT-8208544   \n",
       "1  uniprotkb:Q72547  uniprotkb:Q72547  intact:EBI-7484755|intact:MINT-8208544   \n",
       "2  uniprotkb:Q72547  uniprotkb:Q72547  intact:EBI-7484755|intact:MINT-8208544   \n",
       "3  uniprotkb:Q90VU7  uniprotkb:Q900A7  intact:EBI-7460704|intact:MINT-5281653   \n",
       "4  uniprotkb:Q2Q067  uniprotkb:Q13015                      intact:EBI-9675545   \n",
       "\n",
       "                                       3   \\\n",
       "0  intact:EBI-7484755|intact:MINT-8208544   \n",
       "1  intact:EBI-7484755|intact:MINT-8208544   \n",
       "2  intact:EBI-7484755|intact:MINT-8208544   \n",
       "3  intact:EBI-7460739|intact:MINT-8208185   \n",
       "4                      intact:EBI-6269719   \n",
       "\n",
       "                                                  4   \\\n",
       "0  psi-mi:q72547_9hiv1(display_long)|uniprotkb:po...   \n",
       "1  psi-mi:q72547_9hiv1(display_long)|uniprotkb:po...   \n",
       "2  psi-mi:q72547_9hiv1(display_long)|uniprotkb:po...   \n",
       "3  psi-mi:q90vu7_9hiv1(display_long)|uniprotkb:ne...   \n",
       "4  psi-mi:q2q067_9dela(display_long)|uniprotkb:HB...   \n",
       "\n",
       "                                                  5   \\\n",
       "0  psi-mi:q72547_9hiv1(display_long)|uniprotkb:po...   \n",
       "1  psi-mi:q72547_9hiv1(display_long)|uniprotkb:po...   \n",
       "2  psi-mi:q72547_9hiv1(display_long)|uniprotkb:po...   \n",
       "3  psi-mi:q900a7_9hiv1(display_long)|uniprotkb:ta...   \n",
       "4  psi-mi:af1q_human(display_long)|uniprotkb:MLLT...   \n",
       "\n",
       "                                          6                         7   \\\n",
       "0      psi-mi:\"MI:0030\"(cross-linking study)  Nishitsuji et al. (2011)   \n",
       "1      psi-mi:\"MI:0030\"(cross-linking study)  Nishitsuji et al. (2011)   \n",
       "2      psi-mi:\"MI:0030\"(cross-linking study)  Nishitsuji et al. (2011)   \n",
       "3  psi-mi:\"MI:0416\"(fluorescence microscopy)    Sugiyama et al. (2011)   \n",
       "4         psi-mi:\"MI:0397\"(two hybrid array)     Simonis et al. (2012)   \n",
       "\n",
       "                                                  8   \\\n",
       "0  pubmed:22004763|imex:IM-16791|mint:MINT-820698...   \n",
       "1  pubmed:22004763|imex:IM-16791|mint:MINT-820698...   \n",
       "2  pubmed:22004763|imex:IM-16791|mint:MINT-820698...   \n",
       "3  pubmed:21970979|imex:IM-16787|mint:MINT-820620...   \n",
       "4                      imex:IM-22977|pubmed:22458338   \n",
       "\n",
       "                                                  9   \\\n",
       "0  taxid:11676(9hiv1)|taxid:11676(Human immunodef...   \n",
       "1  taxid:11676(9hiv1)|taxid:11676(Human immunodef...   \n",
       "2  taxid:11676(9hiv1)|taxid:11676(Human immunodef...   \n",
       "3  taxid:11676(9hiv1)|taxid:11676(Human immunodef...   \n",
       "4  taxid:11908(humt-)|taxid:11908(Human T-lymphot...   \n",
       "\n",
       "                                                  10  \\\n",
       "0  taxid:11676(9hiv1)|taxid:11676(Human immunodef...   \n",
       "1  taxid:11676(9hiv1)|taxid:11676(Human immunodef...   \n",
       "2  taxid:11676(9hiv1)|taxid:11676(Human immunodef...   \n",
       "3  taxid:11676(9hiv1)|taxid:11676(Human immunodef...   \n",
       "4         taxid:9606(human)|taxid:9606(Homo sapiens)   \n",
       "\n",
       "                                       11                      12  \\\n",
       "0  psi-mi:\"MI:0915\"(physical association)  psi-mi:\"MI:0471\"(MINT)   \n",
       "1  psi-mi:\"MI:0915\"(physical association)  psi-mi:\"MI:0471\"(MINT)   \n",
       "2  psi-mi:\"MI:0915\"(physical association)  psi-mi:\"MI:0471\"(MINT)   \n",
       "3        psi-mi:\"MI:0403\"(colocalization)  psi-mi:\"MI:0471\"(MINT)   \n",
       "4  psi-mi:\"MI:0915\"(physical association)  psi-mi:\"MI:0471\"(MINT)   \n",
       "\n",
       "                                                  13                   14  \n",
       "0  intact:EBI-7484777|mint:MINT-8206988|imex:IM-1...  intact-miscore:0.40  \n",
       "1  intact:EBI-7484916|mint:MINT-8207003|imex:IM-1...  intact-miscore:0.40  \n",
       "2  intact:EBI-7484965|mint:MINT-8207016|imex:IM-1...  intact-miscore:0.40  \n",
       "3  intact:EBI-7460732|mint:MINT-8206271|imex:IM-1...  intact-miscore:0.27  \n",
       "4                 intact:EBI-9675551|imex:IM-22977-4  intact-miscore:0.49  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mint = pd.read_csv('../data/network/species_human.txt', sep='\\t', header=None)\n",
    "\n",
    "# drop empty interaction columns\n",
    "mint = mint[mint[0].apply(lambda x: ':' in x)]\n",
    "mint = mint[mint[1].apply(lambda x: ':' in x)]\n",
    "\n",
    "data_sources['mint'] = mint\n",
    "data_preproc['mint'] = {\n",
    "    'columns': [0,1],\n",
    "    'maps':[\n",
    "        lambda x: id_dict(x.split(':')[1]),\n",
    "        lambda x: id_dict(x.split(':')[1])\n",
    "    ]\n",
    "}\n",
    "\n",
    "print(mint.columns)\n",
    "mint.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['protein1', 'protein2', 'neighborhood', 'neighborhood_transferred',\n",
      "       'fusion', 'cooccurence', 'homology', 'coexpression',\n",
      "       'coexpression_transferred', 'experiments', 'experiments_transferred',\n",
      "       'database', 'database_transferred', 'textmining',\n",
      "       'textmining_transferred', 'combined_score'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein1</th>\n",
       "      <th>protein2</th>\n",
       "      <th>neighborhood</th>\n",
       "      <th>neighborhood_transferred</th>\n",
       "      <th>fusion</th>\n",
       "      <th>cooccurence</th>\n",
       "      <th>homology</th>\n",
       "      <th>coexpression</th>\n",
       "      <th>coexpression_transferred</th>\n",
       "      <th>experiments</th>\n",
       "      <th>experiments_transferred</th>\n",
       "      <th>database</th>\n",
       "      <th>database_transferred</th>\n",
       "      <th>textmining</th>\n",
       "      <th>textmining_transferred</th>\n",
       "      <th>combined_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>9606.ENSP00000000233</td>\n",
       "      <td>9606.ENSP00000432568</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>62</td>\n",
       "      <td>0</td>\n",
       "      <td>63</td>\n",
       "      <td>900</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>94</td>\n",
       "      <td>909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>9606.ENSP00000000233</td>\n",
       "      <td>9606.ENSP00000427900</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>131</td>\n",
       "      <td>900</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>56</td>\n",
       "      <td>910</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>9606.ENSP00000000233</td>\n",
       "      <td>9606.ENSP00000354878</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>131</td>\n",
       "      <td>900</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>56</td>\n",
       "      <td>910</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>9606.ENSP00000000233</td>\n",
       "      <td>9606.ENSP00000405926</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>63</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>900</td>\n",
       "      <td>0</td>\n",
       "      <td>64</td>\n",
       "      <td>58</td>\n",
       "      <td>906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>9606.ENSP00000000233</td>\n",
       "      <td>9606.ENSP00000314615</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>61</td>\n",
       "      <td>63</td>\n",
       "      <td>0</td>\n",
       "      <td>262</td>\n",
       "      <td>900</td>\n",
       "      <td>0</td>\n",
       "      <td>473</td>\n",
       "      <td>317</td>\n",
       "      <td>971</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                protein1              protein2  neighborhood  \\\n",
       "15  9606.ENSP00000000233  9606.ENSP00000432568             0   \n",
       "16  9606.ENSP00000000233  9606.ENSP00000427900             0   \n",
       "40  9606.ENSP00000000233  9606.ENSP00000354878             0   \n",
       "52  9606.ENSP00000000233  9606.ENSP00000405926             0   \n",
       "78  9606.ENSP00000000233  9606.ENSP00000314615             0   \n",
       "\n",
       "    neighborhood_transferred  fusion  cooccurence  homology  coexpression  \\\n",
       "15                         0       0            0         0             0   \n",
       "16                         0       0            0         0             0   \n",
       "40                         0       0            0         0             0   \n",
       "52                         0       0            0         0             0   \n",
       "78                         0       0            0         0            61   \n",
       "\n",
       "    coexpression_transferred  experiments  experiments_transferred  database  \\\n",
       "15                        62            0                       63       900   \n",
       "16                         0            0                      131       900   \n",
       "40                         0            0                      131       900   \n",
       "52                        63            0                        0       900   \n",
       "78                        63            0                      262       900   \n",
       "\n",
       "    database_transferred  textmining  textmining_transferred  combined_score  \n",
       "15                     0           0                      94             909  \n",
       "16                     0           0                      56             910  \n",
       "40                     0           0                      56             910  \n",
       "52                     0          64                      58             906  \n",
       "78                     0         473                     317             971  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Int64Index([0, 1, 2, 3, 4], dtype='int64')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9606</td>\n",
       "      <td>Q03135|CAV1_HUMAN</td>\n",
       "      <td>9606.ENSP00000339191</td>\n",
       "      <td>100.0</td>\n",
       "      <td>370.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9606</td>\n",
       "      <td>P35125|UBP6_HUMAN</td>\n",
       "      <td>9606.ENSP00000460380</td>\n",
       "      <td>100.0</td>\n",
       "      <td>2941.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9606</td>\n",
       "      <td>P41161|ETV5_HUMAN</td>\n",
       "      <td>9606.ENSP00000306894</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1045.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9606</td>\n",
       "      <td>Q6KF10|GDF6_HUMAN</td>\n",
       "      <td>9606.ENSP00000287020</td>\n",
       "      <td>100.0</td>\n",
       "      <td>928.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9606</td>\n",
       "      <td>Q495C1|RN212_HUMAN</td>\n",
       "      <td>9606.ENSP00000389709</td>\n",
       "      <td>100.0</td>\n",
       "      <td>610.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      0                   1                     2      3       4\n",
       "0  9606   Q03135|CAV1_HUMAN  9606.ENSP00000339191  100.0   370.0\n",
       "1  9606   P35125|UBP6_HUMAN  9606.ENSP00000460380  100.0  2941.0\n",
       "2  9606   P41161|ETV5_HUMAN  9606.ENSP00000306894  100.0  1045.0\n",
       "3  9606   Q6KF10|GDF6_HUMAN  9606.ENSP00000287020  100.0   928.0\n",
       "4  9606  Q495C1|RN212_HUMAN  9606.ENSP00000389709  100.0   610.0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "string_database_score_filter = 900\n",
    "\n",
    "string = pd.read_csv('../data/network/9606.protein.links.full.v11.0.txt', sep=' ')\n",
    "string = string[string['database']>=string_database_score_filter]\n",
    "string_alias = pd.read_csv('../data/network/human.uniprot_2_string.2018.tsv', sep='\\t', header=None)\n",
    "\n",
    "string_to_uniprot = dict(zip(string_alias[2], string_alias[1].apply(lambda x: x.split('|')[1].split('_')[0])))\n",
    "\n",
    "data_sources['string'] = string\n",
    "data_preproc['string'] = {\n",
    "    'columns': ['protein1','protein2'],\n",
    "    'maps': [\n",
    "        lambda x: id_dict(string_to_uniprot[x]) if x in string_to_uniprot.keys() else None,\n",
    "        lambda x: id_dict(string_to_uniprot[x]) if x in string_to_uniprot.keys() else None\n",
    "    ]\n",
    "}\n",
    "\n",
    "print(string.columns)\n",
    "display(string.head())\n",
    "print(string_alias.columns)\n",
    "display(string_alias.head())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Network Assembly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Node1</th>\n",
       "      <th>Node2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Node1, Node2]\n",
       "Index: []"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "network = pd.DataFrame(columns=['Node1', 'Node2'])\n",
    "network.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Before string: 4599 edges (400 self-self)\n",
    "#### No filter on string: 39520 edges\n",
    "#### Database >= 900 filter on string: 5062\n",
    "#### Database >= 800 filter on string: 5294\n",
    "#### Database >= 600 filter on string: 5478"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Adding hippie to network\n",
      "Adding phosphosite_substrate to network\n",
      "Adding hprd to network\n",
      "Adding reactome to network\n",
      "Adding biogrid to network\n",
      "Adding i2d to network\n",
      "Adding mint to network\n",
      "Adding string to network\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Node1</th>\n",
       "      <th>Node2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MST1R</td>\n",
       "      <td>YES1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TYRO3</td>\n",
       "      <td>YES1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>FGR</td>\n",
       "      <td>SRC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ATR</td>\n",
       "      <td>FLT1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>LYN</td>\n",
       "      <td>PRKCD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>LYN</td>\n",
       "      <td>SYK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>LYN</td>\n",
       "      <td>MAP4K1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>PDGFRB</td>\n",
       "      <td>PIK3CA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>PHKG1</td>\n",
       "      <td>PHKG2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>PDGFRA</td>\n",
       "      <td>PDGFRB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>EIF2AK2</td>\n",
       "      <td>PDGFRB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>ARAF</td>\n",
       "      <td>MAP2K1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>RPS6KA4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>KSR2</td>\n",
       "      <td>MAP2K1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>ZAP70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>PTK2B</td>\n",
       "      <td>ZAP70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>MAP2K2</td>\n",
       "      <td>MAPK1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>MAPK1</td>\n",
       "      <td>RPS6KA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>DYRK1B</td>\n",
       "      <td>MAPK1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>KSR2</td>\n",
       "      <td>MAPK1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>PRKCA</td>\n",
       "      <td>SRC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>MYLK</td>\n",
       "      <td>PRKCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>BMPR2</td>\n",
       "      <td>PRKCB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>MAPK6</td>\n",
       "      <td>PRKCB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>GSK3B</td>\n",
       "      <td>PRKCB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>PDPK1</td>\n",
       "      <td>PRKCB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>HK2</td>\n",
       "      <td>PRKCE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>PIK3CB</td>\n",
       "      <td>PRKCE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>PDPK1</td>\n",
       "      <td>PRKCE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>PIK3CB</td>\n",
       "      <td>PRKCD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13739</th>\n",
       "      <td>CAMK1D</td>\n",
       "      <td>PRKACG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13757</th>\n",
       "      <td>PIP4K2C</td>\n",
       "      <td>STK11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13760</th>\n",
       "      <td>TTN</td>\n",
       "      <td>YES1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13764</th>\n",
       "      <td>TTN</td>\n",
       "      <td>WNK1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13765</th>\n",
       "      <td>TRIB2</td>\n",
       "      <td>WEE1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13768</th>\n",
       "      <td>AURKB</td>\n",
       "      <td>CHEK2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13769</th>\n",
       "      <td>AURKB</td>\n",
       "      <td>CDK2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13772</th>\n",
       "      <td>AURKB</td>\n",
       "      <td>CAMKK2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13776</th>\n",
       "      <td>IP6K3</td>\n",
       "      <td>PDPK1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13779</th>\n",
       "      <td>CDK15</td>\n",
       "      <td>STK4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13783</th>\n",
       "      <td>CAMKK2</td>\n",
       "      <td>WNK1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13802</th>\n",
       "      <td>EIF2AK1</td>\n",
       "      <td>TGFBR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13814</th>\n",
       "      <td>HIPK2</td>\n",
       "      <td>WNK1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13820</th>\n",
       "      <td>MKNK2</td>\n",
       "      <td>TGFBR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13824</th>\n",
       "      <td>DYRK4</td>\n",
       "      <td>RPS6KA5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13825</th>\n",
       "      <td>IRAK4</td>\n",
       "      <td>ZAP70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13829</th>\n",
       "      <td>MARK1</td>\n",
       "      <td>STK4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13831</th>\n",
       "      <td>EIF2AK4</td>\n",
       "      <td>PGK1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13832</th>\n",
       "      <td>EIF2AK4</td>\n",
       "      <td>PFKL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13833</th>\n",
       "      <td>EIF2AK4</td>\n",
       "      <td>MAPK3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13837</th>\n",
       "      <td>EIF2AK4</td>\n",
       "      <td>GSK3B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13838</th>\n",
       "      <td>EIF2AK4</td>\n",
       "      <td>PRKAA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13839</th>\n",
       "      <td>EIF2AK4</td>\n",
       "      <td>MAPK14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13859</th>\n",
       "      <td>CAMK2A</td>\n",
       "      <td>RAF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13865</th>\n",
       "      <td>DYRK1B</td>\n",
       "      <td>MAPK11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13872</th>\n",
       "      <td>MAP3K4</td>\n",
       "      <td>MAPK1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13877</th>\n",
       "      <td>PDPK1</td>\n",
       "      <td>ULK1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13879</th>\n",
       "      <td>FLT3</td>\n",
       "      <td>FYN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13881</th>\n",
       "      <td>FLT3</td>\n",
       "      <td>LCK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13884</th>\n",
       "      <td>FLT3</td>\n",
       "      <td>YES1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5062 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Node1    Node2\n",
       "2        MST1R     YES1\n",
       "3        TYRO3     YES1\n",
       "11         FGR      SRC\n",
       "16         ATR     FLT1\n",
       "19         LYN    PRKCD\n",
       "22         LYN      SYK\n",
       "29         LYN   MAP4K1\n",
       "30      PDGFRB   PIK3CA\n",
       "34       PHKG1    PHKG2\n",
       "36      PDGFRA   PDGFRB\n",
       "37     EIF2AK2   PDGFRB\n",
       "45        ARAF   MAP2K1\n",
       "50      MAP2K1  RPS6KA4\n",
       "52        KSR2   MAP2K1\n",
       "58        ABL1    ZAP70\n",
       "59       PTK2B    ZAP70\n",
       "65      MAP2K2    MAPK1\n",
       "66       MAPK1  RPS6KA1\n",
       "70      DYRK1B    MAPK1\n",
       "74        KSR2    MAPK1\n",
       "78       PRKCA      SRC\n",
       "80        MYLK    PRKCA\n",
       "83       BMPR2    PRKCB\n",
       "84       MAPK6    PRKCB\n",
       "85       GSK3B    PRKCB\n",
       "86       PDPK1    PRKCB\n",
       "91         HK2    PRKCE\n",
       "92      PIK3CB    PRKCE\n",
       "93       PDPK1    PRKCE\n",
       "99      PIK3CB    PRKCD\n",
       "...        ...      ...\n",
       "13739   CAMK1D   PRKACG\n",
       "13757  PIP4K2C    STK11\n",
       "13760      TTN     YES1\n",
       "13764      TTN     WNK1\n",
       "13765    TRIB2     WEE1\n",
       "13768    AURKB    CHEK2\n",
       "13769    AURKB     CDK2\n",
       "13772    AURKB   CAMKK2\n",
       "13776    IP6K3    PDPK1\n",
       "13779    CDK15     STK4\n",
       "13783   CAMKK2     WNK1\n",
       "13802  EIF2AK1   TGFBR1\n",
       "13814    HIPK2     WNK1\n",
       "13820    MKNK2   TGFBR1\n",
       "13824    DYRK4  RPS6KA5\n",
       "13825    IRAK4    ZAP70\n",
       "13829    MARK1     STK4\n",
       "13831  EIF2AK4     PGK1\n",
       "13832  EIF2AK4     PFKL\n",
       "13833  EIF2AK4    MAPK3\n",
       "13837  EIF2AK4    GSK3B\n",
       "13838  EIF2AK4   PRKAA2\n",
       "13839  EIF2AK4   MAPK14\n",
       "13859   CAMK2A     RAF1\n",
       "13865   DYRK1B   MAPK11\n",
       "13872   MAP3K4    MAPK1\n",
       "13877    PDPK1     ULK1\n",
       "13879     FLT3      FYN\n",
       "13881     FLT3      LCK\n",
       "13884     FLT3     YES1\n",
       "\n",
       "[5062 rows x 2 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "network = pd.DataFrame(columns=['Node1', 'Node2'])\n",
    "\n",
    "for source_name, data in data_sources.items():\n",
    "    \n",
    "    #if source_name in ['hippie', 'phosphosite_substrate', 'hprd', 'biogrid']:\n",
    "    #    continue\n",
    "    \n",
    "    print('Adding', source_name, 'to network')\n",
    "    \n",
    "    left_col, right_col = data_preproc[source_name]['columns']\n",
    "    left_map, right_map = data_preproc[source_name]['maps']\n",
    "    \n",
    "    data = data[[left_col, right_col]].dropna(axis=0)\n",
    "    \n",
    "    #print(data.dtypes)#[left_col])\n",
    "    \n",
    "    to_add = pd.DataFrame(columns=network.columns)\n",
    "    to_add['Node1'] = data[left_col].apply(left_map)\n",
    "    to_add['Node2'] = data[right_col].apply(right_map)\n",
    "    to_add.dropna(axis=0, inplace = True)\n",
    "   \n",
    "    network = pd.concat([network, to_add], ignore_index=True)\n",
    "    \n",
    "    #break\n",
    "    \n",
    "# drop obvious duplicates\n",
    "network.drop_duplicates(inplace = True)\n",
    "\n",
    "# drop items not in the target kinome set\n",
    "network = network[(network['Node1'].isin(kinome_set)) & (network['Node2'].isin(kinome_set))]\n",
    "\n",
    "network_cop = copy(network[['Node2','Node1']])\n",
    "network_cop.columns = ['Node1', 'Node2']\n",
    "\n",
    "# drop a -> b b -> a redundancies \n",
    "network = network.append(network_cop, ignore_index=True).drop_duplicates()\n",
    "network = network[network['Node1']<network['Node2']]\n",
    "\n",
    "#for i in network.index:\n",
    "#    network.loc[i] = sorted(network.loc[i])\n",
    "    \n",
    "#network.drop_duplicates(inplace=True)\n",
    "\n",
    "network"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Full Comparison"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "understudied = pd.read_csv('../data/dark_kinases.csv', sep=',')\n",
    "understudied.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "understudied_set = set(understudied.symbol)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "kyla_network = pd.read_csv('../data/KIN_edges_no_weights.txt', header=None, sep='\\t')\n",
    "\n",
    "rna_to_MS = kmast.set_index('RNAseq Gene')['MS Gene'].to_dict()\n",
    "hgnc_mapper.update({'KPCA':'PRKCA','KSYK':'SYK', 'VGFR2':'KDR', 'KPCT':'PRKCQ'})\n",
    "kyla_network[0] = kyla_network[0].apply(lambda x: hgnc_mapper[x] if x in hgnc_mapper.keys() else hgnc_mapper[rna_to_MS[x]])\n",
    "kyla_network[1] = kyla_network[1].apply(lambda x: hgnc_mapper[x] if x in hgnc_mapper.keys() else hgnc_mapper[rna_to_MS[x]])\n",
    "\n",
    "kyla_network"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_edges = set(zip(network['Node1'], network['Node2']))\n",
    "kyla_edges = set(zip(kyla_network[0], kyla_network[1]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print('kyla: ', len(kyla_edges))\n",
    "print('new: ', len(new_edges))\n",
    "print('overlap: ',len(new_edges & kyla_edges))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_tot = set(network['Node1']) | set(network['Node2'])\n",
    "kyla_tot = set(kyla_network[0]) | set(kyla_network[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print('kyla: ', len(kyla_tot))\n",
    "print('new: ', len(new_tot))\n",
    "print(len(new_hippie_tot & kyla_tot))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(set(kyla_network[0].unique()) | set(kyla_network[1].unique()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "np.sum(kyla_network[0] == kyla_network[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in kyla_network.index:\n",
    "    kyla_network.loc[i] = sorted(kyla_network.loc[i])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "kyla_tuples = list(zip(kyla_network[0], kyla_network[1]))\n",
    "new_tuples = list(zip(network['Node1'], network['Node2']))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### This used to be > 2700"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Best: 2805"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## missing is now 272"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "missing = set(kyla_tuples) - set(new_tuples)\n",
    "l1, l2 = zip(*missing)\n",
    "problem_kin = list(l1)+list(l2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "set(problem_kin) & understudied_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "problem_kin"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "network.to_csv('../data/kin_unweighted.csv', sep='\\t', header=None, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import Counter\n",
    "\n",
    "c = Counter(problem_kin)\n",
    "c.most_common()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "from collections import Counter\n",
    "\n",
    "c = Counter(problem_kin)\n",
    "c.most_common()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Weighting Functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:anaconda3]",
   "language": "python",
   "name": "conda-env-anaconda3-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
